<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Import sequencing data into R</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R Biovigilance of Phytopathogens Based on a Metabarcoding Approach</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="setup_requirements.html">Workshop Requirements</a>
</li>
<li>
  <a href="1_import_data.html">Importing data into R</a>
</li>
<li>
  <a href="1_import_data.html">Data preprocessing</a>
</li>
<li>
  <a href="1_import_data.html">Data cleaning</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hvanderheyden/Plant_Canada_2024_workshop.github.io">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Import sequencing data into R</h1>

</div>


<p>There are many ways to import data into R, and numerous packages are
available to analyze metabarcoding data. Among these, <a
href="http://www.r-project.org">Phyloseq</a> remains a versatile tool to
import, store and analyze data. Moreover, many other packages can use a
Phyloseq object as a starting point. We’ll start by importing the tables
using the Phyloseq package.</p>
<p>For these steps wee will need the following libraries:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;tidyverse&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span></code></pre></div>
<div id="tables-to-phyloseq" class="section level2">
<h2>Tables to Phyloseq</h2>
<p>Let’s start by importing the tables from the raw_data directory: -
using the read.csv function import the table; - check the first rows and
columns with the head() function; - define the row names from the OTU
column with tibble::column-to_rownames; - transform otu and taxonomy
tables into matrixes (and confirm the object class with the class()
function)</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>biom_illumina<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Raw_data/Illumina/illumina_ASV_table.tsv&quot;</span>, </span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>                        <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">head</span>(biom_illumina[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>],<span class="dv">10</span>)</span></code></pre></div>
<pre><code>##                                 otu QC0221WK20 QC0221WK21 QC0221WK22 QC0221WK23
## 1  f556cceb09c61e1530d00b7d1aed8234          0          0          0          0
## 2  353c06adc7d2edfcfaebf59bbe3ff76e        400          0       2617          0
## 3  9b00bc511c2b9d7b833604813b0b05f2        510       1678       1870       6213
## 4  9e1dc7d16af91d5c34643b33f8f13694          0          0          0          0
## 5  d87923a926982434c811a4fc33a7942f        435       3938        574          0
## 6  9ed6ed511e00f993f42bc338c94d55eb       7513       5636       1616       2387
## 7  d11f971c2573dded4a7b813736d54cc3          0          0        930          0
## 8  a62e14ddf957e10799c77e23e1df70e9          0          0       1098       2531
## 9  1d05a08941ce5f54c3ecac0a1d850780        951       4696        932        865
## 10 da68b84ae7440726f0fa022c707fe95b          0          0          0          0</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>biom_illumina <span class="ot">&lt;-</span> biom_illumina <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;otu&quot;</span>) </span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>biom_illumina <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(biom_illumina)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="fu">class</span>(biom_illumina)</span></code></pre></div>
<pre><code>## [1] &quot;matrix&quot; &quot;array&quot;</code></pre>
<p>Next, import the taxonomy table in the same way.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>taxo_illumina<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Raw_data/Illumina/illumina_taxonomy.tsv&quot;</span>, </span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>                        <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="fu">head</span>(taxo_illumina[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],<span class="dv">10</span>)</span></code></pre></div>
<pre><code>##                                 otu      Kingdom      Phylum            Class
## 1  f556cceb09c61e1530d00b7d1aed8234 k__Eukaryota p__Oomycota c__Stramenopiles
## 2  353c06adc7d2edfcfaebf59bbe3ff76e k__Eukaryota p__Oomycota c__Stramenopiles
## 3  9b00bc511c2b9d7b833604813b0b05f2 k__Eukaryota p__Oomycota c__Stramenopiles
## 4  9e1dc7d16af91d5c34643b33f8f13694 k__Eukaryota p__Oomycota c__Stramenopiles
## 5  d87923a926982434c811a4fc33a7942f k__Eukaryota p__Oomycota c__Stramenopiles
## 6  9ed6ed511e00f993f42bc338c94d55eb k__Eukaryota p__Oomycota c__Stramenopiles
## 7  d11f971c2573dded4a7b813736d54cc3 k__Eukaryota p__Oomycota c__Stramenopiles
## 8  a62e14ddf957e10799c77e23e1df70e9 k__Eukaryota p__Oomycota c__Stramenopiles
## 9  1d05a08941ce5f54c3ecac0a1d850780 k__Eukaryota p__Oomycota c__Stramenopiles
## 10 da68b84ae7440726f0fa022c707fe95b k__Eukaryota p__Oomycota c__Stramenopiles
##                Order             Family               Genus
## 1  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 2  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 3  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 4  o__Peronosporales f__Peronosporaceae           g__Bremia
## 5     o__Albuginales    f__Albuginaceae           g__Albugo
## 6  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 7  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 8  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 9  o__Peronosporales f__Peronosporaceae g__Hyaloperonospora
## 10 o__Peronosporales f__Peronosporaceae g__Hyaloperonospora
##                          Species
## 1      s__Peronospora_manshurica
## 2      s__Peronospora_destructor
## 3      s__Peronospora_variabilis
## 4            s__Bremia_elliptica
## 5              s__Albugo_candida
## 6        s__Peronospora_aparines
## 7                               
## 8      s__Peronospora_variabilis
## 9    s__Hyaloperonospora_nesliae
## 10 s__Hyaloperonospora_brassicae</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>taxo_illumina <span class="ot">&lt;-</span> taxo_illumina <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;otu&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>taxo_illumina <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(taxo_illumina)</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="fu">class</span>(taxo_illumina) </span></code></pre></div>
<pre><code>## [1] &quot;matrix&quot; &quot;array&quot;</code></pre>
<p>And merge the two tables into a single phyloseq objects</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>OTU_illumina <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(biom_illumina, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>TAX_illumina <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(taxo_illumina)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>illuminaPS <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(OTU_illumina, TAX_illumina)</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>illuminaPS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 324 taxa and 112 samples ]
## tax_table()   Taxonomy Table:    [ 324 taxa by 7 taxonomic ranks ]</code></pre>
<p>Let’s repeat for the IonTorrent dataset</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>biom_IonTorrent<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Raw_data/IonTorrent/iontorrent_ASV_table.tsv&quot;</span>, </span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>                        <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="fu">head</span>(biom_IonTorrent[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>],<span class="dv">10</span>)</span></code></pre></div>
<pre><code>##                                 otu IonT_QC0221WK20 IonT_QC0221WK23
## 1  2949b355cd00039476d9a808f2e76b8e               0               0
## 2  9d34744d5f40b8a99cc028bb4908e00b              16              27
## 3  61c226ba90e400b577fe1d486db3256d               7               0
## 4  af77980eb4e8e0abee3729db262bcdae               3               0
## 5  5f838c2663e51903bc2edc67cf5dde72               0               0
## 6  7a77fe76bb62ff1e46fe2906e1906041               0               0
## 7  7b3fb79d8a7c200c7b15d79ba9fb1f6b               0           33270
## 8  582d1b304d60b4c074d550f3d08af368               0               0
## 9  865d613b55499cf3db6e4477a885676c               0               0
## 10 70351fcb4a655bd2db93506501ca7533               0               0
##    IonT_QC0321WK23 IonT_QC0421WK23
## 1               48              11
## 2                0               0
## 3                0               0
## 4                8               0
## 5                0               0
## 6                6               0
## 7               14            8989
## 8                0               0
## 9                0            4574
## 10               0               0</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>biom_IonTorrent <span class="ot">&lt;-</span> biom_IonTorrent <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;otu&quot;</span>)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>biom_IonTorrent <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(biom_IonTorrent)</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="fu">class</span>(biom_IonTorrent)</span></code></pre></div>
<pre><code>## [1] &quot;matrix&quot; &quot;array&quot;</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>taxo_IonTorrent<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Raw_data/IonTorrent/iontorrent_taxonomy.tsv&quot;</span>, </span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>                        <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a><span class="fu">head</span>(taxo_IonTorrent[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],<span class="dv">10</span>)</span></code></pre></div>
<pre><code>##                                 otu      Kingdom      Phylum            Class
## 1  2949b355cd00039476d9a808f2e76b8e k__Eukaryota p__Oomycota c__Stramenopiles
## 2  9d34744d5f40b8a99cc028bb4908e00b k__Eukaryota p__Oomycota c__Stramenopiles
## 3  61c226ba90e400b577fe1d486db3256d k__Eukaryota p__Oomycota c__Stramenopiles
## 4  af77980eb4e8e0abee3729db262bcdae k__Eukaryota p__Oomycota c__Stramenopiles
## 5  5f838c2663e51903bc2edc67cf5dde72 k__Eukaryota p__Oomycota c__Stramenopiles
## 6  7a77fe76bb62ff1e46fe2906e1906041 k__Eukaryota p__Oomycota c__Stramenopiles
## 7  7b3fb79d8a7c200c7b15d79ba9fb1f6b k__Eukaryota p__Oomycota c__Stramenopiles
## 8  582d1b304d60b4c074d550f3d08af368 k__Eukaryota p__Oomycota c__Stramenopiles
## 9  865d613b55499cf3db6e4477a885676c k__Eukaryota p__Oomycota c__Stramenopiles
## 10 70351fcb4a655bd2db93506501ca7533 k__Eukaryota p__Oomycota c__Stramenopiles
##                Order             Family               Genus
## 1  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 2  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 3  o__Peronosporales f__Peronosporaceae           g__Bremia
## 4  o__Peronosporales f__Peronosporaceae g__Hyaloperonospora
## 5  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 6  o__Peronosporales f__Peronosporaceae       g__Plasmopara
## 7  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 8  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 9  o__Peronosporales f__Peronosporaceae      g__Peronospora
## 10 o__Peronosporales f__Peronosporaceae g__Hyaloperonospora
##                        Species
## 1    s__Peronospora_manshurica
## 2    s__Peronospora_manshurica
## 3          s__Bremia_elliptica
## 4  s__Hyaloperonospora_nesliae
## 5    s__Peronospora_manshurica
## 6       s__Plasmopara_viticola
## 7    s__Peronospora_variabilis
## 8    s__Peronospora_manshurica
## 9  s__Peronospora_boni-henrici
## 10</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>taxo_IonTorrent <span class="ot">&lt;-</span> taxo_IonTorrent <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;otu&quot;</span>) </span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>taxo_IonTorrent <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(taxo_IonTorrent)</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a><span class="fu">class</span>(taxo_IonTorrent) </span></code></pre></div>
<pre><code>## [1] &quot;matrix&quot; &quot;array&quot;</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>OTU_IonTorrent <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(biom_IonTorrent, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>TAX_IonTorrent <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(taxo_IonTorrent)</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>IonTorrentPS <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(OTU_IonTorrent, TAX_IonTorrent)</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>IonTorrentPS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1225 taxa and 94 samples ]
## tax_table()   Taxonomy Table:    [ 1225 taxa by 7 taxonomic ranks ]</code></pre>
<p>And Again for the Nanopore dataset</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>biom_Nanopore<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Raw_data/Nanopore/Nanopore_pseudo_ASV_table.csv&quot;</span>, </span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>                        <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">&quot;;&quot;</span>)</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="fu">head</span>(biom_Nanopore[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>],<span class="dv">10</span>)</span></code></pre></div>
<pre><code>##           otu plaque_1_its_B1 plaque_1_its_B10 plaque_1_its_B11
## 1  MW633475.1            3394             4832            14977
## 2  AF347031.1            2367              716             3042
## 3  MZ159329.1               8                0                0
## 4  MH855943.1            3759              481             4916
## 5  KU850565.1              19               12               68
## 6  MW709918.1             112                1                1
## 7  MF924721.1               0                0                1
## 8  MG828953.1             458               92              187
## 9  MT310617.1               2                6                0
## 10 JN383490.1               0                0                1
##    plaque_1_its_B12
## 1              1582
## 2               908
## 3                 0
## 4               711
## 5                49
## 6                 2
## 7                 0
## 8                29
## 9                 3
## 10                3</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>biom_Nanopore <span class="ot">&lt;-</span> biom_Nanopore <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;otu&quot;</span>)</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>biom_Nanopore <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(biom_Nanopore)</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a><span class="fu">class</span>(biom_Nanopore)</span></code></pre></div>
<pre><code>## [1] &quot;matrix&quot; &quot;array&quot;</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>taxo_Nanopore<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Raw_data/Nanopore/Nanopore_taxonomy.csv&quot;</span>, </span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>                        <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">&quot;;&quot;</span>)</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="fu">head</span>(taxo_Nanopore[,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],<span class="dv">10</span>)</span></code></pre></div>
<pre><code>##           otu      Kingdom           Phylum             Class             Order
## 1  MW633475.1 k__Eukaryota    p__Ascomycota c__dothideomyceta o__Cladosporiales
## 2  AF347031.1 k__Eukaryota    p__Ascomycota c__dothideomyceta   o__Pleosporales
## 3  MZ159329.1 k__Eukaryota p__Basidiomycota   c__Opisthokonta    o__Corticiales
## 4  MH855943.1 k__Eukaryota    p__Ascomycota c__dothideomyceta   o__Pleosporales
## 5  JX089579.1 k__Eukaryota    p__Ascomycota c__dothideomyceta   o__Pleosporales
## 6  JN906978.1 k__Eukaryota    p__Ascomycota c__dothideomyceta o__Cladosporiales
## 7  KU850565.1 k__Eukaryota    p__Ascomycota c__dothideomyceta   o__Pleosporales
## 8  MW709918.1 k__Eukaryota p__Basidiomycota   c__Opisthokonta    o__Tremellales
## 9  MF924721.1 k__Eukaryota p__Basidiomycota   c__Opisthokonta    o__Polyporales
## 10 MG828953.1 k__Eukaryota    p__Ascomycota c__dothideomyceta   o__Pleosporales
##                   Family               Genus                        Species
## 1     f__Cladosporiaceae     g__Cladosporium       s__Cladosporium_herbarum
## 2       f__Pleosporaceae       g__Alternaria        s__Alternaria_alternata
## 3        f__Corticiaceae       g__Phlebiella             s__Phlebiella_vaga
## 4       f__Didymellaceae        g__Epicoccum            s__Epicoccum_nigrum
## 5       f__Pleosporaceae        g__Bipolaris       s__Bipolaris_microstegii
## 6     f__Cladosporiaceae     g__Cladosporium        s__Cladosporium_aphidis
## 7       f__Pleosporaceae      g__Stemphylium      s__Stemphylium_vesicarium
## 8         f__Bulleraceae          g__Bullera                s__Bullera_alba
## 9   f__Incrustoporiaceae     g__Skeletocutis s__Skeletocutis_mopanshanensis
## 10 f__Didymosphaeriaceae g__Pseudopithomyces      s__Pseudopithomyces_rosae</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>taxo_Nanopore <span class="ot">&lt;-</span> taxo_Nanopore <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;otu&quot;</span>) </span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>taxo_Nanopore <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(taxo_Nanopore)</span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a><span class="fu">class</span>(taxo_Nanopore) </span></code></pre></div>
<pre><code>## [1] &quot;matrix&quot; &quot;array&quot;</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>OTU_Nanopore <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(biom_Nanopore, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>TAX_Nanopore <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(taxo_Nanopore)</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>NanoporePS <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(OTU_Nanopore, TAX_Nanopore)</span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>NanoporePS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 4241 taxa and 94 samples ]
## tax_table()   Taxonomy Table:    [ 4241 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="merging-objects" class="section level2">
<h2>Merging objects</h2>
<p>Now we want to merge the phyloseq objects we’ve created To do that we
will generate: - a combined biom-table (using the phyloseq::otu_table
function) - a combined taxonomy table (using the
phyloseq::tax_table)</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>otu_illumina <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(illuminaPS)</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>otu_Iontorrent <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(IonTorrentPS)</span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>otu_Nanopore <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(NanoporePS)</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>combined_otu <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(otu_illumina, otu_Iontorrent, otu_Nanopore)</span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>tax_illumina  <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(illuminaPS)</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>tax_Iontorrent <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(IonTorrentPS)</span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>tax_Nanopore <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(NanoporePS)</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>combined_tax <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(tax_illumina, tax_Iontorrent, tax_Nanopore)</span></code></pre></div>
</div>
<div id="adding-metadata" class="section level2">
<h2>Adding metadata</h2>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Raw_data/metadata_organicsoil.tsv&quot;</span>, </span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>                 <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a><span class="fu">head</span>(meta)</span></code></pre></div>
<pre><code>##          SampleID              InputFileName   Samplename   seq_tech SiteID
## 1 IonT_QC0221WK20  B1_9_L001_R1_001.fastq.gz QC02_21_WK20 IonTorrent   QC02
## 2 IonT_QC0221WK21  B2_9_L001_R1_001.fastq.gz QC02_21_WK21 IonTorrent   QC02
## 3 IonT_QC0221WK22  B4_9_L001_R1_001.fastq.gz QC02_21_WK22 IonTorrent   QC02
## 4 IonT_QC0221WK23 B10_9_L001_R1_001.fastq.gz QC02_21_WK23 IonTorrent   QC02
## 5 IonT_QC0221WK24 B16_9_L001_R1_001.fastq.gz QC02_21_WK24 IonTorrent   QC02
## 6 IonT_QC0221WK25 B22_9_L001_R1_001.fastq.gz QC02_21_WK25 IonTorrent   QC02
##         Region Year Week  X_coord  Y_coord
## 1 Frelighsburg 2021 WK20 45.05353 -72.8625
## 2 Frelighsburg 2021 WK21 45.05353 -72.8625
## 3 Frelighsburg 2021 WK22 45.05353 -72.8625
## 4 Frelighsburg 2021 WK23 45.05353 -72.8625
## 5 Frelighsburg 2021 WK24 45.05353 -72.8625
## 6 Frelighsburg 2021 WK25 45.05353 -72.8625</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span> </span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a></span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>samples <span class="ot">=</span> <span class="fu">sample_data</span>(meta)</span></code></pre></div>
<p>Finally, merge the biom table, tax table and metadata into a final
phyloseq object:</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>combinedPS <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(combined_otu, combined_tax, samples);combinedPS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 5790 taxa and 282 samples ]
## sample_data() Sample Data:       [ 282 samples by 9 sample variables ]
## tax_table()   Taxonomy Table:    [ 5790 taxa by 7 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ape&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Attachement du package : &#39;ape&#39;</code></pre>
<pre><code>## L&#39;objet suivant est masqué depuis &#39;package:dplyr&#39;:
## 
##     where</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>random_tree_combinedPS <span class="ot">=</span> <span class="fu">rtree</span>(<span class="fu">ntaxa</span>(combinedPS), </span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>                    <span class="at">rooted=</span><span class="cn">TRUE</span>, </span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>                    <span class="at">tip.label=</span><span class="fu">taxa_names</span>(combinedPS))</span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a></span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a>combinedPS_final <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(combined_otu, combined_tax, samples, random_tree_combinedPS)</span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a>combinedPS_final</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 5790 taxa and 282 samples ]
## sample_data() Sample Data:       [ 282 samples by 9 sample variables ]
## tax_table()   Taxonomy Table:    [ 5790 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 5790 tips and 5789 internal nodes ]</code></pre>
</div>
<div id="saving-phyloseq-objects" class="section level2">
<h2>Saving phyloseq objects</h2>
<p>Before saving our work, to make sure that everything is under
control, you can inspect the phyloseq object with the following
functions:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="fu">sample_names</span>(combinedPS_final)</span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a><span class="fu">rank_names</span>(combinedPS_final)</span>
<span id="cb44-3"><a href="#cb44-3" tabindex="-1"></a><span class="fu">sample_variables</span>(combinedPS_final)</span></code></pre></div>
<p>Finally, we save our phyloseq object into the processed_data
directory</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">saveRDS</span>(combinedPS_final, <span class="at">file=</span> <span class="st">&quot;Processed_data/combinedPS_final.rds&quot;</span>)</span></code></pre></div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
