<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>stacked bar charts</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R Biovigilance of Phytopathogens Based on a Metabarcoding Approach</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="setup_requirements.html">Workshop Requirements</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part I: Data cleaning and filtering
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1_import_data.html">Importing data into R</a>
    </li>
    <li>
      <a href="2_preprocessing.html">Data cleaning</a>
    </li>
    <li>
      <a href="3_summaries.html">Data filtering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part II -- Comparing sequencing techniques
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4_stacked_barchart.html">stacked bar charts</a>
    </li>
    <li>
      <a href="5_comparison.html">Comparisons</a>
    </li>
  </ul>
</li>
<li>
  <a href="6_Heat_map.html">Beta diversity and comaprison</a>
</li>
<li>
  <a href="7_Time_series.html">Temporal variation</a>
</li>
<li>
  <a href="8_biomarker-discovery.html">Biomarker-discovery</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hvanderheyden/Plant_Canada_2024_workshop.github.io">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">stacked bar charts</h1>

</div>


<p>One of the first things that comes to mind when presenting
metabarcoding data is the ever-popular stacked bar chart. Again, there
are many packages that can be used to draw stacked bar charts (see for
example <a
href="https://github.com/YuLab-SMU/MicrobiotaProcess">MicrobiotaProcess</a>),
but let’s do it manually this time instead.</p>
<p>Let’s load our latest Phyloseq object and librariries.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;tidyverse&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>WorkingPS<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;Processed_data/combinedPS_clean_pruned_filtered.rds&quot;</span>);WorkingPS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 2969 taxa and 277 samples ]
## sample_data() Sample Data:       [ 277 samples by 11 sample variables ]
## tax_table()   Taxonomy Table:    [ 2969 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 2969 tips and 2968 internal nodes ]</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># Because The Nanopore data set also contains fungi, lets keep only the Oomycota Phylum. </span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>WorkingPS<span class="ot">&lt;-</span><span class="fu">subset_taxa</span>(WorkingPS, Phylum<span class="sc">==</span><span class="st">&quot;Oomycota&quot;</span>);WorkingPS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1491 taxa and 277 samples ]
## sample_data() Sample Data:       [ 277 samples by 11 sample variables ]
## tax_table()   Taxonomy Table:    [ 1491 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1491 tips and 1490 internal nodes ]</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">saveRDS</span>(WorkingPS, <span class="at">file=</span> <span class="st">&quot;Processed_data/WorkingPS.rds&quot;</span>)</span></code></pre></div>
<p>Here is a function borrowed from the MicrobiotaProcess package that
will be helpful later when plotting our bar chart.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>get_cols <span class="ot">&lt;-</span> <span class="cf">function</span> (n){</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#8dd3c7&quot;</span>, <span class="st">&quot;#ffffb3&quot;</span>, <span class="st">&quot;#bebada&quot;</span>, <span class="st">&quot;#fb8072&quot;</span>, <span class="st">&quot;#80b1d3&quot;</span>,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>           <span class="st">&quot;#fdb462&quot;</span>, <span class="st">&quot;#b3de69&quot;</span>, <span class="st">&quot;#fccde5&quot;</span>, <span class="st">&quot;#d9d9d9&quot;</span>, <span class="st">&quot;#bc80bd&quot;</span>,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>           <span class="st">&quot;#ccebc5&quot;</span>, <span class="st">&quot;#ffed6f&quot;</span>)</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  </span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  col2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#1f78e4&quot;</span>, <span class="st">&quot;#ffff33&quot;</span>, <span class="st">&quot;#c2a5cf&quot;</span>, <span class="st">&quot;#ff7f00&quot;</span>, <span class="st">&quot;#810f7c&quot;</span>,</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>            <span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#006d2c&quot;</span>, <span class="st">&quot;#4d4d4d&quot;</span>, <span class="st">&quot;#8c510a&quot;</span>, <span class="st">&quot;#d73027&quot;</span>,</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>            <span class="st">&quot;#78c679&quot;</span>, <span class="st">&quot;#7f0000&quot;</span>, <span class="st">&quot;#41b6c4&quot;</span>, <span class="st">&quot;#e7298a&quot;</span>, <span class="st">&quot;#54278f&quot;</span>)</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>  </span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>  col3 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#1f78b4&quot;</span>, <span class="st">&quot;#b2df8a&quot;</span>, <span class="st">&quot;#33a02c&quot;</span>, <span class="st">&quot;#fb9a99&quot;</span>,</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>            <span class="st">&quot;#e31a1c&quot;</span>, <span class="st">&quot;#fdbf6f&quot;</span>, <span class="st">&quot;#ff7f00&quot;</span>, <span class="st">&quot;#cab2d6&quot;</span>, <span class="st">&quot;#6a3d9a&quot;</span>,</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>            <span class="st">&quot;#ffff99&quot;</span>, <span class="st">&quot;#b15928&quot;</span>)</span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>  <span class="fu">colorRampPalette</span>(col2)(n)</span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>}</span></code></pre></div>
<div id="summarize-the-data" class="section level2">
<h2>Summarize the data</h2>
<p>The phyloseq psmelt function converts our phyloseq object into a data
frame.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>WorkingPS_melt <span class="ot">&lt;-</span> <span class="fu">psmelt</span>(WorkingPS)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="fu">head</span>(WorkingPS_melt)</span></code></pre></div>
<pre><code>##                                     OTU          Sample Abundance
## 61883  2949b355cd00039476d9a808f2e76b8e IonT_QC0621WK33    390300
## 61974  2949b355cd00039476d9a808f2e76b8e IonT_QC0421WK33    389752
## 246778 9d34744d5f40b8a99cc028bb4908e00b IonT_QC0421WK32    332168
## 246676 9d34744d5f40b8a99cc028bb4908e00b IonT_QC0621WK34    323922
## 246690 9d34744d5f40b8a99cc028bb4908e00b IonT_QC0521WK33    302992
## 61805  2949b355cd00039476d9a808f2e76b8e IonT_QC0621WK32    258720
##                     InputFileName   Samplename   seq_tech SiteID      Region
## 61883  B82_9_L001_R1_001.fastq.gz QC06_21_WK33 IonTorrent   QC06 Sherrington
## 61974  B80_9_L001_R1_001.fastq.gz QC04_21_WK33 IonTorrent   QC04 Napierville
## 246778 B73_9_L001_R1_001.fastq.gz QC04_21_WK32 IonTorrent   QC04 Napierville
## 246676 B89_9_L001_R1_001.fastq.gz QC06_21_WK34 IonTorrent   QC06 Sherrington
## 246690 B81_9_L001_R1_001.fastq.gz QC05_21_WK33 IonTorrent   QC05  St-Edouard
## 61805  B75_9_L001_R1_001.fastq.gz QC06_21_WK32 IonTorrent   QC06 Sherrington
##        Year Week Week2       Date  X_coord  Y_coord   Kingdom   Phylum
## 61883  2021 WK33    33 2021-08-09 45.13174 -73.4719 Eukaryota Oomycota
## 61974  2021 WK33    33 2021-08-09 45.18664 -73.3426 Eukaryota Oomycota
## 246778 2021 WK32    32 2021-08-02 45.18664 -73.3426 Eukaryota Oomycota
## 246676 2021 WK34    34 2021-08-16 45.13174 -73.4719 Eukaryota Oomycota
## 246690 2021 WK33    33 2021-08-09 45.19820 -73.5421 Eukaryota Oomycota
## 61805  2021 WK32    32 2021-08-02 45.13174 -73.4719 Eukaryota Oomycota
##                Class          Order          Family       Genus
## 61883  Stramenopiles Peronosporales Peronosporaceae Peronospora
## 61974  Stramenopiles Peronosporales Peronosporaceae Peronospora
## 246778 Stramenopiles Peronosporales Peronosporaceae Peronospora
## 246676 Stramenopiles Peronosporales Peronosporaceae Peronospora
## 246690 Stramenopiles Peronosporales Peronosporaceae Peronospora
## 61805  Stramenopiles Peronosporales Peronosporaceae Peronospora
##                       Species
## 61883  Peronospora manshurica
## 61974  Peronospora manshurica
## 246778 Peronospora manshurica
## 246676 Peronospora manshurica
## 246690 Peronospora manshurica
## 61805  Peronospora manshurica</code></pre>
<p>Using the dplyr package (included in the tidyverse package) we create
a tibble to summarize abundance and use the pivot_longer function to
transpose the taxonomy from column to row.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>genus_abundance <span class="ot">&lt;-</span> WorkingPS_melt <span class="sc">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_tech, Week) <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rel_abund =</span> Abundance <span class="sc">/</span> <span class="fu">sum</span>(Abundance)) <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Abundance) <span class="sc">%&gt;%</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>),</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">&quot;level&quot;</span>,</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">&quot;taxon&quot;</span>)</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="fu">str</span>(genus_abundance)</span></code></pre></div>
<pre><code>## tibble [2,891,049 × 16] (S3: tbl_df/tbl/data.frame)
##  $ OTU          : chr [1:2891049] &quot;2949b355cd00039476d9a808f2e76b8e&quot; &quot;2949b355cd00039476d9a808f2e76b8e&quot; &quot;2949b355cd00039476d9a808f2e76b8e&quot; &quot;2949b355cd00039476d9a808f2e76b8e&quot; ...
##  $ Sample       : chr [1:2891049] &quot;IonT_QC0621WK33&quot; &quot;IonT_QC0621WK33&quot; &quot;IonT_QC0621WK33&quot; &quot;IonT_QC0621WK33&quot; ...
##  $ InputFileName: chr [1:2891049] &quot;B82_9_L001_R1_001.fastq.gz&quot; &quot;B82_9_L001_R1_001.fastq.gz&quot; &quot;B82_9_L001_R1_001.fastq.gz&quot; &quot;B82_9_L001_R1_001.fastq.gz&quot; ...
##  $ Samplename   : chr [1:2891049] &quot;QC06_21_WK33&quot; &quot;QC06_21_WK33&quot; &quot;QC06_21_WK33&quot; &quot;QC06_21_WK33&quot; ...
##  $ seq_tech     : chr [1:2891049] &quot;IonTorrent&quot; &quot;IonTorrent&quot; &quot;IonTorrent&quot; &quot;IonTorrent&quot; ...
##  $ SiteID       : chr [1:2891049] &quot;QC06&quot; &quot;QC06&quot; &quot;QC06&quot; &quot;QC06&quot; ...
##  $ Region       : chr [1:2891049] &quot;Sherrington&quot; &quot;Sherrington&quot; &quot;Sherrington&quot; &quot;Sherrington&quot; ...
##  $ Year         : int [1:2891049] 2021 2021 2021 2021 2021 2021 2021 2021 2021 2021 ...
##  $ Week         : chr [1:2891049] &quot;WK33&quot; &quot;WK33&quot; &quot;WK33&quot; &quot;WK33&quot; ...
##  $ Week2        : int [1:2891049] 33 33 33 33 33 33 33 33 33 33 ...
##  $ Date         : chr [1:2891049] &quot;2021-08-09&quot; &quot;2021-08-09&quot; &quot;2021-08-09&quot; &quot;2021-08-09&quot; ...
##  $ X_coord      : num [1:2891049] 45.1 45.1 45.1 45.1 45.1 ...
##  $ Y_coord      : num [1:2891049] -73.5 -73.5 -73.5 -73.5 -73.5 ...
##  $ rel_abund    : num [1:2891049] 0.192 0.192 0.192 0.192 0.192 ...
##  $ level        : chr [1:2891049] &quot;Kingdom&quot; &quot;Phylum&quot; &quot;Class&quot; &quot;Order&quot; ...
##  $ taxon        : chr [1:2891049] &quot;Eukaryota&quot; &quot;Oomycota&quot; &quot;Stramenopiles&quot; &quot;Peronosporales&quot; ...</code></pre>
</div>
<div id="genus-bar-chart" class="section level2">
<h2>Genus bar chart</h2>
<p>We then summarize the data, by sequencing technology and by week, at
the genus level and input them into ggplot2 to obtain our bar chart with
weeks on the x-axis and relative abundance on the y-axis, with facets
for sequencing technologies.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>genus_abundance <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="fu">filter</span>(level<span class="sc">==</span><span class="st">&quot;Genus&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_tech, Week, taxon) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rel_abund =</span> <span class="fu">sum</span>(rel_abund), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_tech, Week, taxon) <span class="sc">%&gt;%</span> </span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_rel_abund=</span><span class="fu">mean</span>(rel_abund), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Week, <span class="at">y=</span>mean_rel_abund, <span class="at">fill=</span>taxon))<span class="sc">+</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>  <span class="fu">geom_col</span>()<span class="sc">+</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(seq_tech), <span class="at">nrow =</span> <span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Week&quot;</span>,</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>       <span class="at">y=</span><span class="st">&quot;Relative abundance (%)&quot;</span>)<span class="sc">+</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&quot;italic&quot;</span>))<span class="sc">+</span></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">get_cols</span>(<span class="dv">29</span>))<span class="sc">+</span> <span class="co"># This line calls the &quot;get_cols&quot;&gt;&quot; function above</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span> <span class="fu">guide_legend</span>(<span class="at">keywidth =</span> <span class="fl">0.6</span>, </span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a>                            <span class="at">keyheight =</span> <span class="fl">0.7</span>, </span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a>                            <span class="at">ncol=</span><span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">60</span>, </span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a>                                   <span class="at">vjust =</span> <span class="fl">0.5</span>, </span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a>                                   <span class="at">hjust=</span><span class="fl">0.4</span>))</span></code></pre></div>
<p><img src="4_stacked_barchart_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="species-bar-chart" class="section level2">
<h2>Species bar chart</h2>
<p>The first bar chart shows that the profiles are fairly similar
between the three technologies, although some genera appear to be over-
or under-represented with certain technologies. We would now like to
examine the distribution of species as a function of time and sequencing
technology. However, with over 80 species, it is difficult to get a
clear representation of species distribution with a stacked bar chart,
and we need to reduce the number of species to be plotted.</p>
<p>To do that let’s start by creating a list of predefined species that
we want to be plotted in our bar chart:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>predefined_species <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Albugo candida&quot;</span>,</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>                       <span class="st">&quot;Bremia lactucae cluster&quot;</span>,</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora nesliae&quot;</span>,</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora brassicae&quot;</span>,</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora camelinae&quot;</span>,</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>                       <span class="st">&quot;Peronospora manshurica&quot;</span>,</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>                       <span class="st">&quot;Peronospora variabilis&quot;</span>,</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>                       <span class="st">&quot;Peronospora destructor&quot;</span>,                       </span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>                       <span class="st">&quot;Peronospora aparines&quot;</span>,</span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>                       <span class="st">&quot;Peronospora erucastri&quot;</span>,</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>                       <span class="st">&quot;Plasmopara viticola&quot;</span>,</span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a>                       <span class="st">&quot;Pseudoperonospora cubensis&quot;</span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>)</span></code></pre></div>
<p>And replace anything not in our list with “Other”:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>WorkingPS_melt2 <span class="ot">&lt;-</span>  WorkingPS_melt <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">case_when</span>(</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>    Species <span class="sc">%in%</span> </span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>      predefined_species <span class="sc">~</span> Species,  <span class="co"># Keep valid species as is</span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Other&quot;</span>  <span class="co"># Replace other species with &quot;other&quot;</span></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>  ))</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a><span class="fu">head</span>(WorkingPS_melt2)</span></code></pre></div>
<pre><code>##                                     OTU          Sample Abundance
## 61883  2949b355cd00039476d9a808f2e76b8e IonT_QC0621WK33    390300
## 61974  2949b355cd00039476d9a808f2e76b8e IonT_QC0421WK33    389752
## 246778 9d34744d5f40b8a99cc028bb4908e00b IonT_QC0421WK32    332168
## 246676 9d34744d5f40b8a99cc028bb4908e00b IonT_QC0621WK34    323922
## 246690 9d34744d5f40b8a99cc028bb4908e00b IonT_QC0521WK33    302992
## 61805  2949b355cd00039476d9a808f2e76b8e IonT_QC0621WK32    258720
##                     InputFileName   Samplename   seq_tech SiteID      Region
## 61883  B82_9_L001_R1_001.fastq.gz QC06_21_WK33 IonTorrent   QC06 Sherrington
## 61974  B80_9_L001_R1_001.fastq.gz QC04_21_WK33 IonTorrent   QC04 Napierville
## 246778 B73_9_L001_R1_001.fastq.gz QC04_21_WK32 IonTorrent   QC04 Napierville
## 246676 B89_9_L001_R1_001.fastq.gz QC06_21_WK34 IonTorrent   QC06 Sherrington
## 246690 B81_9_L001_R1_001.fastq.gz QC05_21_WK33 IonTorrent   QC05  St-Edouard
## 61805  B75_9_L001_R1_001.fastq.gz QC06_21_WK32 IonTorrent   QC06 Sherrington
##        Year Week Week2       Date  X_coord  Y_coord   Kingdom   Phylum
## 61883  2021 WK33    33 2021-08-09 45.13174 -73.4719 Eukaryota Oomycota
## 61974  2021 WK33    33 2021-08-09 45.18664 -73.3426 Eukaryota Oomycota
## 246778 2021 WK32    32 2021-08-02 45.18664 -73.3426 Eukaryota Oomycota
## 246676 2021 WK34    34 2021-08-16 45.13174 -73.4719 Eukaryota Oomycota
## 246690 2021 WK33    33 2021-08-09 45.19820 -73.5421 Eukaryota Oomycota
## 61805  2021 WK32    32 2021-08-02 45.13174 -73.4719 Eukaryota Oomycota
##                Class          Order          Family       Genus
## 61883  Stramenopiles Peronosporales Peronosporaceae Peronospora
## 61974  Stramenopiles Peronosporales Peronosporaceae Peronospora
## 246778 Stramenopiles Peronosporales Peronosporaceae Peronospora
## 246676 Stramenopiles Peronosporales Peronosporaceae Peronospora
## 246690 Stramenopiles Peronosporales Peronosporaceae Peronospora
## 61805  Stramenopiles Peronosporales Peronosporaceae Peronospora
##                       Species
## 61883  Peronospora manshurica
## 61974  Peronospora manshurica
## 246778 Peronospora manshurica
## 246676 Peronospora manshurica
## 246690 Peronospora manshurica
## 61805  Peronospora manshurica</code></pre>
<p>Generate a second tibble with using this predefined list of
species:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>genus_abundance2 <span class="ot">&lt;-</span> WorkingPS_melt2 <span class="sc">%&gt;%</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_tech, Week) <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rel_abund =</span> Abundance <span class="sc">/</span> <span class="fu">sum</span>(Abundance)) <span class="sc">%&gt;%</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Abundance) <span class="sc">%&gt;%</span></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>),</span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">&quot;level&quot;</span>,</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">&quot;taxon&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">taxon =</span> <span class="fu">factor</span>(taxon,</span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a>                       <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Albugo candida&quot;</span>,</span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a>                       <span class="st">&quot;Bremia lactucae cluster&quot;</span>,</span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora nesliae&quot;</span>,</span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora brassicae&quot;</span>,</span>
<span id="cb15-14"><a href="#cb15-14" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora camelinae&quot;</span>,</span>
<span id="cb15-15"><a href="#cb15-15" tabindex="-1"></a>                       <span class="st">&quot;Peronospora manshurica&quot;</span>,</span>
<span id="cb15-16"><a href="#cb15-16" tabindex="-1"></a>                       <span class="st">&quot;Peronospora variabilis&quot;</span>,</span>
<span id="cb15-17"><a href="#cb15-17" tabindex="-1"></a>                       <span class="st">&quot;Peronospora destructor&quot;</span>,                       </span>
<span id="cb15-18"><a href="#cb15-18" tabindex="-1"></a>                       <span class="st">&quot;Peronospora aparines&quot;</span>,</span>
<span id="cb15-19"><a href="#cb15-19" tabindex="-1"></a>                       <span class="st">&quot;Peronospora erucastri&quot;</span>,</span>
<span id="cb15-20"><a href="#cb15-20" tabindex="-1"></a>                       <span class="st">&quot;Plasmopara viticola&quot;</span>,</span>
<span id="cb15-21"><a href="#cb15-21" tabindex="-1"></a>                       <span class="st">&quot;Pseudoperonospora cubensis&quot;</span>,</span>
<span id="cb15-22"><a href="#cb15-22" tabindex="-1"></a>                       <span class="st">&quot;Other&quot;</span>)))</span></code></pre></div>
<p>summarize the data at the species level and pipe that into ggplot2 to
get our bar chart at the species level:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>genus_abundance2 <span class="sc">%&gt;%</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>  <span class="fu">filter</span>(level<span class="sc">==</span><span class="st">&quot;Species&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_tech, Week, taxon) <span class="sc">%&gt;%</span> </span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rel_abund =</span> <span class="fu">sum</span>(rel_abund), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_tech, Week, taxon) <span class="sc">%&gt;%</span> </span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_rel_abund=</span><span class="fu">mean</span>(rel_abund), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Week, <span class="at">y=</span>mean_rel_abund, <span class="at">fill=</span>taxon))<span class="sc">+</span></span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>  <span class="fu">geom_col</span>()<span class="sc">+</span></span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(seq_tech), <span class="at">nrow =</span> <span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Week&quot;</span>,</span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a>       <span class="at">y=</span><span class="st">&quot;Relative abundance (%)&quot;</span>)<span class="sc">+</span></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&quot;italic&quot;</span>))<span class="sc">+</span></span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">get_cols</span>(<span class="dv">14</span>))<span class="sc">+</span> <span class="co"># This line calls the &quot;get_cols&quot;&gt;&quot; function above</span></span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span> <span class="fu">guide_legend</span>(<span class="at">keywidth =</span> <span class="fl">0.6</span>, </span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>                            <span class="at">keyheight =</span> <span class="fl">0.7</span>, </span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a>                            <span class="at">ncol=</span><span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">60</span>, </span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a>                                   <span class="at">vjust =</span> <span class="fl">0.5</span>, </span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a>                                   <span class="at">hjust=</span><span class="fl">0.4</span>))</span></code></pre></div>
<p><img src="4_stacked_barchart_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>Again we can see some similitude and differences between sequencing
technologies, now lets see how different they are.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
