<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Sequencing technology comparison</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R Biovigilance of Phytopathogens Based on a Metabarcoding Approach</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="setup_requirements.html">Workshop Requirements</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part I: Data cleaning and filtering
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1_import_data.html">Importing data into R</a>
    </li>
    <li>
      <a href="2_preprocessing.html">Data cleaning</a>
    </li>
    <li>
      <a href="3_summaries.html">Data filtering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part II -- Comparing sequencing techniques
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4_stacked_barchart.html">stacked bar charts</a>
    </li>
    <li>
      <a href="5_comparison.html">Comparisons</a>
    </li>
  </ul>
</li>
<li>
  <a href="6_Heat_map.html">Part III -- Anlysing diversity</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hvanderheyden/Plant_Canada_2024_workshop.github.io">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Supplementary material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1_import_data.html">Heat maps</a>
    </li>
    <li>
      <a href="2_preprocessing.html">alpha diversity</a>
    </li>
    <li>
      <a href="3_summaries.html">Temporal distribution</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Sequencing technology comparison</h1>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;MicrobiotaProcess&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ape&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;UpSetR&quot;</span>)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;vegan&quot;</span>)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>WorkingPS<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;Processed_data/WorkingPS.rds&quot;</span>)</span></code></pre></div>
<p>To compare our sequencing methods, we first need to use the taxonomy;
we can’t make the comparison at the ASV level since they are not
constant across data sets.</p>
<p>The tax_glom function from the phyloseq package is used to aggregate
data with the same taxonomy.The get_taxadf function from the
MicrobiotaProcess package also performs this task.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>WorkingPS_glom <span class="ot">&lt;-</span> <span class="fu">tax_glom</span>(WorkingPS, <span class="at">taxrank=</span><span class="st">&quot;Species&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">saveRDS</span>(WorkingPS_glom, <span class="at">file=</span> <span class="st">&quot;Processed_data/WorkingPS_glom.rds&quot;</span>)</span></code></pre></div>
<div id="pca" class="section level2">
<h2>PCA</h2>
<p>We can then use our agglomerated taxonomy to compare our 3 sequencing
methods using principal component analysis (PCA).</p>
<p>Once again, we’ll use the MicrobiotaProcess package to obtain and
plot the results of our PCA.</p>
<ul>
<li>Note here that we have used the robust centered log ratio (rclr), a
transformation method commonly used in compositional data analysis.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>WorkingPS_glom_PCA <span class="ot">&lt;-</span> <span class="fu">get_pca</span>(<span class="at">obj=</span>WorkingPS_glom, </span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>                               <span class="at">method=</span>(<span class="st">&quot;rclr&quot;</span>))</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>pcaplot1 <span class="ot">&lt;-</span><span class="fu">ggordpoint</span>(<span class="at">obj=</span>WorkingPS_glom_PCA, <span class="at">pc=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">biplot=</span><span class="cn">FALSE</span>, <span class="at">speciesannot=</span><span class="cn">FALSE</span>,</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>                      <span class="at">factorNames=</span><span class="fu">c</span>(<span class="st">&quot;seq_tech&quot;</span>), <span class="at">ellipse=</span><span class="cn">TRUE</span>)<span class="sc">+</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">get_cols</span>(<span class="dv">6</span>))<span class="sc">+</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;Null&quot;</span>)</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>pcaplot2 <span class="ot">&lt;-</span> <span class="fu">ggordpoint</span>(<span class="at">obj=</span>WorkingPS_glom_PCA, <span class="at">pc=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>), <span class="at">biplot=</span><span class="cn">FALSE</span>, <span class="at">speciesannot=</span><span class="cn">FALSE</span>,</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>                       <span class="at">factorNames=</span><span class="fu">c</span>(<span class="st">&quot;seq_tech&quot;</span>), <span class="at">ellipse=</span><span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">get_cols</span>(<span class="dv">6</span>))</span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="fu">ggarrange</span>(<span class="fu">ggarrange</span>(pcaplot1,</span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>                    pcaplot2,</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>                    <span class="at">common.legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>                    <span class="at">legend =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a>                    <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>),</span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>                    <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">2.5</span>, <span class="dv">1</span>)))</span></code></pre></div>
<p><img src="5_comparison_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>We can see from the graph that the data are fairly clustered by
technology, that nanopore and illumina data are fairly close together,
and that IonTorrent data are more scattered.</p>
<p>Let’s try a principal coordinate analysis (PCoA).</p>
</div>
<div id="pcoa" class="section level2">
<h2>PCOA</h2>
<p>Similarly, let’s use the get_pcoa function (MicrobiotaProcess) to
obtain and plot our PCoA results.</p>
<ul>
<li>Note here that we have used the wieghted unifrac distance method,
which account for phylogenetic distance and abundance.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>pcoa_results <span class="ot">&lt;-</span> <span class="fu">get_pcoa</span>(<span class="at">obj=</span>WorkingPS_glom, </span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>                      <span class="at">distmethod=</span><span class="st">&quot;wunifrac&quot;</span>, </span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>                      <span class="at">method=</span><span class="st">&quot;hellinger&quot;</span>)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>pcoaplot1 <span class="ot">&lt;-</span> <span class="fu">ggordpoint</span>(<span class="at">obj=</span>pcoa_results,</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>                          <span class="at">pc=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>),</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>                          <span class="at">biplot=</span><span class="cn">FALSE</span>,</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>                          <span class="at">factorNames=</span><span class="fu">c</span>(<span class="st">&quot;seq_tech&quot;</span>), </span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>                          <span class="at">ellipse=</span><span class="cn">TRUE</span>,</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>                          <span class="at">poinsize =</span> <span class="fl">1.8</span>,</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>                          <span class="at">stroke =</span> <span class="fl">0.2</span>)<span class="sc">+</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  <span class="fu">theme</span> (<span class="at">legend.position =</span> <span class="st">&quot;Null&quot;</span>)<span class="sc">+</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">get_cols</span>(<span class="dv">6</span>))<span class="sc">+</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>pcoaplot2 <span class="ot">&lt;-</span> <span class="fu">ggordpoint</span>(<span class="at">obj=</span>pcoa_results,</span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>                          <span class="at">pc=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>),</span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>                          <span class="at">biplot=</span><span class="cn">FALSE</span>,</span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>                          <span class="at">factorNames=</span><span class="fu">c</span>(<span class="st">&quot;seq_tech&quot;</span>), </span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>                          <span class="at">ellipse=</span><span class="cn">TRUE</span>,</span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a>                          <span class="at">poinsize =</span> <span class="fl">1.8</span>,</span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a>                          <span class="at">stroke =</span> <span class="fl">0.2</span>)<span class="sc">+</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.box =</span> <span class="st">&quot;horizontal&quot;</span>, <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>)<span class="sc">+</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">get_cols</span>(<span class="dv">6</span>))<span class="sc">+</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>))<span class="sc">+</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_blank</span>())<span class="sc">+</span> </span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span>)<span class="sc">+</span></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">keywidth =</span> <span class="fl">0.6</span>, </span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>                             <span class="at">keyheight =</span> <span class="fl">0.7</span>,</span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>                             <span class="at">ncol =</span> <span class="dv">3</span>))</span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb4-33"><a href="#cb4-33" tabindex="-1"></a><span class="fu">ggarrange</span>(<span class="fu">ggarrange</span>(pcoaplot1,</span>
<span id="cb4-34"><a href="#cb4-34" tabindex="-1"></a>                    pcoaplot2,</span>
<span id="cb4-35"><a href="#cb4-35" tabindex="-1"></a>                    <span class="at">common.legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-36"><a href="#cb4-36" tabindex="-1"></a>                    <span class="at">legend =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb4-37"><a href="#cb4-37" tabindex="-1"></a>                    <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>),</span>
<span id="cb4-38"><a href="#cb4-38" tabindex="-1"></a>                    <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">2.5</span>, <span class="dv">1</span>)))</span></code></pre></div>
<p><img src="5_comparison_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>Once again, we can see that the data are grouped by technology, and
that Illumina’s data seem closer to Nanopore’s than IonTorrent’s.</p>
</div>
<div id="upset-plot" class="section level2">
<h2>Upset Plot</h2>
<p>Like Venn diagrams, UpSet graphs are a data visualization method for
quantifying elements shared between and unique to given elements of
comparison.</p>
<p>We use the get_upset function from the MicrobiotaProcess package to
create our figure.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>upsetda <span class="ot">&lt;-</span> <span class="fu">get_upset</span>(<span class="at">obj=</span>WorkingPS_glom, <span class="at">factorNames=</span><span class="st">&quot;seq_tech&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">upset</span>(upsetda, <span class="at">sets=</span><span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">sample_data</span>(WorkingPS_glom)<span class="sc">$</span>seq_tech)), </span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>      <span class="at">sets.bar.color =</span> <span class="st">&quot;#56B4E9&quot;</span>,</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>      <span class="at">order.by =</span> <span class="st">&quot;freq&quot;</span>, </span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>      <span class="at">empty.intersections =</span> <span class="st">&quot;on&quot;</span>)</span></code></pre></div>
<p><img src="5_comparison_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>We can see from this graph that 22 taxa are shared by the 3
technologies, while 35 taxa are unique to IonTorrent and 19 taxa to
Nanopore.</p>
</div>
<div id="permanova" class="section level2">
<h2>Permanova</h2>
<p>We can therefore assume that there are differences between squaring
technologies, but the question remains, is there a significant
difference between the technologies?</p>
<p>Let’s use Permutational Multivariate Analysis of Variance (PERMANOVA)
to determine whether there is such a difference between the
technologies.</p>
<p>We’ll be using the Vegan package for this analysis.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># get the distance matrix </span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>dist_mat <span class="ot">&lt;-</span> <span class="fu">get_dist</span>(WorkingPS_glom, <span class="at">distmethod =</span><span class="st">&quot;wunifrac&quot;</span>, <span class="at">method=</span><span class="st">&quot;hellinger&quot;</span>)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co"># Match the sample names </span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>sampleda <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(WorkingPS_glom), <span class="at">check.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>sampleda <span class="ot">&lt;-</span> sampleda[<span class="fu">match</span>(<span class="fu">colnames</span>(<span class="fu">as.matrix</span>(dist_mat)),<span class="fu">rownames</span>(sampleda)),,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>sampleda<span class="sc">$</span>seq_tech <span class="ot">&lt;-</span> <span class="fu">factor</span>(sampleda<span class="sc">$</span>seq_tech)</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co"># compute the PERMANOVA </span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1024</span>)</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>adonis_res <span class="ot">&lt;-</span> <span class="fu">adonis</span>(dist_mat <span class="sc">~</span> seq_tech<span class="sc">*</span>Week, <span class="at">data=</span>sampleda, <span class="at">permutation=</span><span class="dv">999</span>)</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co"># print the results</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="fu">data.frame</span>(adonis_res<span class="sc">$</span>aov.tab)</span></code></pre></div>
<pre><code>##                Df SumsOfSqs    MeanSqs   F.Model         R2 Pr..F.
## seq_tech        2  8.849188 4.42459423 48.041920 0.17077948  0.001
## Week           15 18.230060 1.21533733 13.196044 0.35181985  0.001
## seq_tech:Week  30  3.646632 0.12155439  1.319829 0.07037593  0.004
## Residuals     229 21.090583 0.09209861        NA 0.40702475     NA
## Total         276 51.816463         NA        NA 1.00000000     NA</code></pre>
<p>So we can see that there are differences between the technologies,
but what are they? Let’s try a pairwise comparison.</p>
<p>First between IonTorrent and Nanopore</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>Ion_nano<span class="ot">&lt;-</span> <span class="fu">subset_samples</span>(WorkingPS_glom, seq_tech <span class="sc">!=</span><span class="st">&quot;Illumina&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>dist_Ion_nano <span class="ot">&lt;-</span> <span class="fu">get_dist</span>(Ion_nano, <span class="at">distmethod =</span><span class="st">&quot;wunifrac&quot;</span>, <span class="at">method=</span><span class="st">&quot;hellinger&quot;</span>)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>sampleda_Ion_nano <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(Ion_nano), <span class="at">check.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>sampleda_Ion_nano <span class="ot">&lt;-</span> sampleda_Ion_nano[<span class="fu">match</span>(<span class="fu">colnames</span>(<span class="fu">as.matrix</span>(dist_Ion_nano)),<span class="fu">rownames</span>(sampleda_Ion_nano)),,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>sampleda_Ion_nano<span class="sc">$</span>seq_tech <span class="ot">&lt;-</span> <span class="fu">factor</span>(sampleda_Ion_nano<span class="sc">$</span>seq_tech)</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1024</span>)</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>adonis_Ion_nano <span class="ot">&lt;-</span> <span class="fu">adonis</span>(dist_Ion_nano <span class="sc">~</span> seq_tech<span class="sc">*</span>Week, </span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>                          <span class="at">data=</span>sampleda_Ion_nano, </span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>                          <span class="at">permutation=</span><span class="dv">999</span>)</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="co"># create an object of type numeric to store the PERMANOVA results </span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>pairwise_p<span class="ot">&lt;-</span><span class="fu">numeric</span>()</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co"># and store the results for each comparison</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>pairwise_p[<span class="st">&quot;Ion_nano&quot;</span>] <span class="ot">&lt;-</span> adonis_Ion_nano[[<span class="st">&quot;aov.tab&quot;</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span></code></pre></div>
<p>Then between IonTorrent and Illumina</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="do">##### Adonis for diseased plantations vs forest  ####</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>Ion_Illu<span class="ot">&lt;-</span> <span class="fu">subset_samples</span>(WorkingPS_glom, seq_tech <span class="sc">!=</span><span class="st">&quot;Nanopore&quot;</span>)</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>dist_Ion_Illu <span class="ot">&lt;-</span> <span class="fu">get_dist</span>(Ion_Illu, <span class="at">distmethod =</span><span class="st">&quot;wunifrac&quot;</span>, <span class="at">method=</span><span class="st">&quot;hellinger&quot;</span>)</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>sampleda_Ion_Illu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(Ion_Illu), <span class="at">check.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>sampleda_Ion_Illu <span class="ot">&lt;-</span> sampleda_Ion_Illu[<span class="fu">match</span>(<span class="fu">colnames</span>(<span class="fu">as.matrix</span>(dist_Ion_Illu)),<span class="fu">rownames</span>(sampleda_Ion_Illu)),,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>sampleda_Ion_Illu<span class="sc">$</span>seq_tech <span class="ot">&lt;-</span> <span class="fu">factor</span>(sampleda_Ion_Illu<span class="sc">$</span>seq_tech)</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1024</span>)</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>adonis_Ion_Illu <span class="ot">&lt;-</span> <span class="fu">adonis</span>(dist_Ion_Illu <span class="sc">~</span> seq_tech<span class="sc">*</span>Week, </span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>                          <span class="at">data=</span>sampleda_Ion_Illu, </span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>                          <span class="at">permutation=</span><span class="dv">999</span>)</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a>pairwise_p[<span class="st">&quot;Ion_Illu&quot;</span>] <span class="ot">&lt;-</span> adonis_Ion_Illu[[<span class="st">&quot;aov.tab&quot;</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span></code></pre></div>
<p>And finally between NAnopore and Illumina</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="do">##### Adonis for diseased plantations vs forest  ####</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>Nano_Illu<span class="ot">&lt;-</span> <span class="fu">subset_samples</span>(WorkingPS_glom, seq_tech <span class="sc">!=</span><span class="st">&quot;IonTorrent&quot;</span>)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>dist_Nano_Illu <span class="ot">&lt;-</span> <span class="fu">get_dist</span>(Nano_Illu, <span class="at">distmethod =</span><span class="st">&quot;wunifrac&quot;</span>, <span class="at">method=</span><span class="st">&quot;hellinger&quot;</span>)</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>sampleda_Nano_Illu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(Nano_Illu), <span class="at">check.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>sampleda_Nano_Illu <span class="ot">&lt;-</span> sampleda_Nano_Illu[<span class="fu">match</span>(<span class="fu">colnames</span>(<span class="fu">as.matrix</span>(dist_Nano_Illu)),<span class="fu">rownames</span>(sampleda_Nano_Illu)),,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>sampleda_Nano_Illu<span class="sc">$</span>seq_tech <span class="ot">&lt;-</span> <span class="fu">factor</span>(sampleda_Nano_Illu<span class="sc">$</span>seq_tech)</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1024</span>)</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>adonis_Nano_Illu <span class="ot">&lt;-</span> <span class="fu">adonis</span>(dist_Nano_Illu <span class="sc">~</span> seq_tech<span class="sc">*</span>Week, </span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>                          <span class="at">data=</span>sampleda_Nano_Illu, </span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>                          <span class="at">permutation=</span><span class="dv">999</span>)</span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a>pairwise_p[<span class="st">&quot;Nano_Illu&quot;</span>] <span class="ot">&lt;-</span> adonis_Nano_Illu[[<span class="st">&quot;aov.tab&quot;</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span></code></pre></div>
<p>When processing pairwise data, it is important to correct the
significance level for multiple comparisons, such as the Bonferroni
correction. So lets apply the Bonferroni correction to our Permanova
results.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">p.adjust</span>(pairwise_p, <span class="at">method=</span><span class="st">&quot;bonferroni&quot;</span>)</span></code></pre></div>
<pre><code>##  Ion_nano  Ion_Illu Nano_Illu 
##     0.003     0.003     0.003</code></pre>
<p>Given these results, we can conclude that the three methods are
different from one another.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">detach</span>(package<span class="sc">:</span>MicrobiotaProcess)</span></code></pre></div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
