---
title: "Sequencing technology comparison"
output: html_document
---

```{r setup, include=FALSE} 
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
```

```{r, message=FALSE,echo=FALSE}
get_cols <- function (n){
  col <- c("#8dd3c7", "#ffffb3", "#bebada", "#fb8072", "#80b1d3",
           "#fdb462", "#b3de69", "#fccde5", "#d9d9d9", "#bc80bd",
           "#ccebc5", "#ffed6f")
  
  col2 <- c("#1f78e4", "#ffff33", "#c2a5cf", "#ff7f00", "#810f7c",
            "#a6cee3", "#006d2c", "#4d4d4d", "#8c510a", "#d73027",
            "#78c679", "#7f0000", "#41b6c4", "#e7298a", "#54278f")
  
  col3 <- c("#a6cee3", "#1f78b4", "#b2df8a", "#33a02c", "#fb9a99",
            "#e31a1c", "#fdbf6f", "#ff7f00", "#cab2d6", "#6a3d9a",
            "#ffff99", "#b15928")
  colorRampPalette(col2)(n)
}
```

```{r, message=FALSE,echo=TRUE}
library("MicrobiotaProcess")
library("ggpubr")
library("ape")
library("UpSetR")
library("vegan")

WorkingPS<-readRDS("Processed_data/WorkingPS.rds")
```

To compare our sequencing methods, we first need to use the taxonomy; we can't make the comparison at the ASV level since they are not constant across data sets. 

The tax_glom function from the phyloseq package is used to aggregate data with the same taxonomy.The get_taxadf function from the MicrobiotaProcess package also performs this task. 

```{r}

WorkingPS_glom <- tax_glom(WorkingPS, taxrank="Species")


saveRDS(WorkingPS_glom, file= "Processed_data/WorkingPS_glom.rds")

```

PCA 
----

We can then use our agglomerated taxonomy to compare our 3 sequencing methods using principal component analysis (PCA). 

Once again, we'll use the MicrobiotaProcess package to obtain and plot the results of our PCA. 

* Note here that we have used the robust centered log ratio (rclr), a transformation method commonly used in compositional data analysis. 

```{r}
WorkingPS_glom_PCA <- get_pca(obj=WorkingPS_glom, 
                               method=("rclr"))


pcaplot1 <-ggordpoint(obj=WorkingPS_glom_PCA, pc=c(1, 2), biplot=FALSE, speciesannot=FALSE,
                      factorNames=c("seq_tech"), ellipse=TRUE)+
  scale_fill_manual(values=get_cols(6))+
    theme(plot.title = element_blank())+
  theme(legend.position="Null")

pcaplot2 <- ggordpoint(obj=WorkingPS_glom_PCA, pc=c(1, 3), biplot=FALSE, speciesannot=FALSE,
                       factorNames=c("seq_tech"), ellipse=TRUE) +
    theme(plot.title = element_blank())+
  scale_fill_manual(values=get_cols(6))

library("ggpubr")
ggarrange(ggarrange(pcaplot1,
                    pcaplot2,
                    common.legend = TRUE,
                    legend = "bottom",
                    ncol = 2, labels = c("A", "B"),
                    heights = c(2.5, 1)))


```


We can see from the graph that the data are fairly clustered by technology, that nanopore and illumina data are fairly close together, and that IonTorrent data are more scattered. 

Let's try a principal coordinate analysis (PCoA).

PCOA 
---- 

Similarly, let's use the get_pcoa function (MicrobiotaProcess) to obtain and plot our PCoA results.

* Note here that we have used the wieghted unifrac distance method, which account for phylogenetic distance and abundance.  

```{r}
pcoa_results <- get_pcoa(obj=WorkingPS_glom, 
                      distmethod="wunifrac", 
                      method="hellinger")

pcoaplot1 <- ggordpoint(obj=pcoa_results,
                          pc=c(1, 2),
                          biplot=FALSE,
                          factorNames=c("seq_tech"), 
                          ellipse=TRUE,
                          poinsize = 1.8,
                          stroke = 0.2)+
  theme (legend.position = "Null")+
  scale_fill_manual(values=get_cols(6))+
  theme(plot.title = element_blank())

pcoaplot2 <- ggordpoint(obj=pcoa_results,
                          pc=c(1, 3),
                          biplot=FALSE,
                          factorNames=c("seq_tech"), 
                          ellipse=TRUE,
                          poinsize = 1.8,
                          stroke = 0.2)+
  theme(legend.box = "horizontal", legend.position = "right")+
  scale_fill_manual(values=get_cols(6))+
  theme(legend.text=element_text(size=8))+
  theme(legend.title = element_blank())+ 
  theme(plot.title = element_blank())+
  theme(legend.direction = "horizontal", legend.box = "vertical")+
  guides(fill = guide_legend(keywidth = 0.6, 
                             keyheight = 0.7,
                             ncol = 3))
library("ggpubr")
ggarrange(ggarrange(pcoaplot1,
                    pcoaplot2,
                    common.legend = TRUE,
                    legend = "bottom",
                    ncol = 2, labels = c("A", "B"),
                    heights = c(2.5, 1)))


```

Once again, we can see that the data are grouped by technology, and that Illumina's data seem closer to Nanopore's than IonTorrent's.  

Upset Plot 
----

Like Venn diagrams, UpSet graphs are a data visualization method for quantifying elements shared between and unique to given elements of comparison.

We use the get_upset function from the MicrobiotaProcess package to create our figure. 

```{r}
upsetda <- get_upset(obj=WorkingPS_glom, factorNames="seq_tech")

upset(upsetda, sets=unique(as.vector(sample_data(WorkingPS_glom)$seq_tech)), 
      sets.bar.color = "#56B4E9",
      order.by = "freq", 
      empty.intersections = "on")

```

We can see from this graph that 22 taxa are shared by the 3 technologies, while 35 taxa are unique to IonTorrent and 19 taxa to Nanopore.  

Permanova
---- 

We can therefore assume that there are differences between squaring technologies, but the question remains, is there a significant difference between the technologies? 

Let's use Permutational Multivariate Analysis of Variance (PERMANOVA) to determine whether there is such a difference between the technologies.  

We'll be using the Vegan package for this analysis.

```{r}
# get the distance matrix 
dist_mat <- get_dist(WorkingPS_glom, distmethod ="wunifrac", method="hellinger")
# Match the sample names 
sampleda <- data.frame(sample_data(WorkingPS_glom), check.names=FALSE)
sampleda <- sampleda[match(colnames(as.matrix(dist_mat)),rownames(sampleda)),,drop=FALSE]
sampleda$seq_tech <- factor(sampleda$seq_tech)
# compute the PERMANOVA 
set.seed(1024)
adonis_res <- adonis(dist_mat ~ seq_tech*Week, data=sampleda, permutation=999)
# print the results
data.frame(adonis_res$aov.tab)

```

So we can see that there are differences between the technologies, but what are they? Let's try a pairwise comparison.


First between IonTorrent and Nanopore 
```{r}
Ion_nano<- subset_samples(WorkingPS_glom, seq_tech !="Illumina")

dist_Ion_nano <- get_dist(Ion_nano, distmethod ="wunifrac", method="hellinger")
sampleda_Ion_nano <- data.frame(sample_data(Ion_nano), check.names=FALSE)
sampleda_Ion_nano <- sampleda_Ion_nano[match(colnames(as.matrix(dist_Ion_nano)),rownames(sampleda_Ion_nano)),,drop=FALSE]
sampleda_Ion_nano$seq_tech <- factor(sampleda_Ion_nano$seq_tech)

set.seed(1024)
adonis_Ion_nano <- adonis(dist_Ion_nano ~ seq_tech*Week, 
                          data=sampleda_Ion_nano, 
                          permutation=999)

# create an object of type numeric to store the PERMANOVA results 
pairwise_p<-numeric()
# and store the results for each comparison
pairwise_p["Ion_nano"] <- adonis_Ion_nano[["aov.tab"]][["Pr(>F)"]][1]
```

Then between IonTorrent and Illumina 
```{r}
##### Adonis for diseased plantations vs forest  ####
Ion_Illu<- subset_samples(WorkingPS_glom, seq_tech !="Nanopore")

dist_Ion_Illu <- get_dist(Ion_Illu, distmethod ="wunifrac", method="hellinger")
sampleda_Ion_Illu <- data.frame(sample_data(Ion_Illu), check.names=FALSE)
sampleda_Ion_Illu <- sampleda_Ion_Illu[match(colnames(as.matrix(dist_Ion_Illu)),rownames(sampleda_Ion_Illu)),,drop=FALSE]
sampleda_Ion_Illu$seq_tech <- factor(sampleda_Ion_Illu$seq_tech)

set.seed(1024)
adonis_Ion_Illu <- adonis(dist_Ion_Illu ~ seq_tech*Week, 
                          data=sampleda_Ion_Illu, 
                          permutation=999)

pairwise_p["Ion_Illu"] <- adonis_Ion_Illu[["aov.tab"]][["Pr(>F)"]][1]
```

And finally between NAnopore and Illumina 
```{r}
##### Adonis for diseased plantations vs forest  ####
Nano_Illu<- subset_samples(WorkingPS_glom, seq_tech !="IonTorrent")

dist_Nano_Illu <- get_dist(Nano_Illu, distmethod ="wunifrac", method="hellinger")
sampleda_Nano_Illu <- data.frame(sample_data(Nano_Illu), check.names=FALSE)
sampleda_Nano_Illu <- sampleda_Nano_Illu[match(colnames(as.matrix(dist_Nano_Illu)),rownames(sampleda_Nano_Illu)),,drop=FALSE]
sampleda_Nano_Illu$seq_tech <- factor(sampleda_Nano_Illu$seq_tech)

set.seed(1024)
adonis_Nano_Illu <- adonis(dist_Nano_Illu ~ seq_tech*Week, 
                          data=sampleda_Nano_Illu, 
                          permutation=999)


pairwise_p["Nano_Illu"] <- adonis_Nano_Illu[["aov.tab"]][["Pr(>F)"]][1]
```

When processing pairwise data, it is important to correct the significance level for multiple comparisons, such as the Bonferroni correction. So lets apply the Bonferroni correction to our Permanova results. 

```{r}
p.adjust(pairwise_p, method="bonferroni")
```

Given these results, we can conclude that the three methods are different from one another.

```{r}
detach(package:MicrobiotaProcess)
```