<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Temporal diversity</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R Biovigilance of Phytopathogens Based on a Metabarcoding Approach</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="setup_requirements.html">Workshop Requirements</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part I: Data cleaning and filtering
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1_import_data.html">Importing data into R</a>
    </li>
    <li>
      <a href="2_preprocessing.html">Data cleaning</a>
    </li>
    <li>
      <a href="3_summaries.html">Data filtering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part II -- Comparing sequencing techniques
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4_stacked_barchart.html">stacked bar charts</a>
    </li>
    <li>
      <a href="5_comparison.html">Comparisons</a>
    </li>
  </ul>
</li>
<li>
  <a href="6_Heat_map.html">Beta diversity and comaprison</a>
</li>
<li>
  <a href="7_Time_series.html">Temporal variation</a>
</li>
<li>
  <a href="8_biomarker-discovery.html">Biomarker-discovery</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hvanderheyden/Plant_Canada_2024_workshop.github.io">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Temporal diversity</h1>

</div>


<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ampvis2&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>WorkingPS_glom<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;Processed_data/WorkingPS_glom.rds&quot;</span>)</span></code></pre></div>
<div id="heat-map-by-week-and-technologies" class="section level2">
<h2>Heat map by week and technologies</h2>
<p>First we need to convert our phyloseq object into an Ampvis2 object,
then generate the heat map with the weeks (times) in x and the
sequencing technology as facets.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Convert the PS object to ampvis </span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>WorkingPS_amp <span class="ot">&lt;-</span> <span class="fu">amp_load</span>(WorkingPS_glom)</span></code></pre></div>
</div>
<div id="time-series" class="section level2">
<h2>Time series</h2>
<p>When data are collected over time, such as our aerobiome data, we may
be tempted to explore the temporal distribution patterns of specific
species. Continuing with the Ampvis2 package, we can use the
amp_timeseries function to plot read distribution over time.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">amp_timeseries</span>(WorkingPS_amp,</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="at">time_variable =</span> <span class="st">&quot;Date&quot;</span>,</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="at">tax_aggregate =</span> <span class="st">&quot;Species&quot;</span>,</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="at">tax_show =</span> <span class="dv">9</span>,</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">normalise=</span><span class="cn">FALSE</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="7_Time_series_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>Or with read counts to be in percent per sample.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">amp_timeseries</span>(WorkingPS_amp,</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  <span class="at">time_variable =</span> <span class="st">&quot;Date&quot;</span>,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">tax_aggregate =</span> <span class="st">&quot;Species&quot;</span>,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="at">tax_show =</span> <span class="dv">9</span>,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="at">normalise=</span><span class="cn">TRUE</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="7_Time_series_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>We can also generate our curves by species in individual graphs.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">amp_timeseries</span>(WorkingPS_amp,</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="at">time_variable =</span> <span class="st">&quot;Date&quot;</span>,</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="co">#group_by = &quot;seq_tech&quot;,</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="at">split =</span> <span class="cn">TRUE</span>,</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>,</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  <span class="at">tax_show =</span> <span class="dv">12</span>,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  <span class="at">tax_aggregate =</span> <span class="st">&quot;Species&quot;</span>,</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  <span class="at">tax_add =</span> <span class="cn">NULL</span>,</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  <span class="at">normalise=</span><span class="cn">TRUE</span>,</span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>  <span class="co">#plotly = TRUE,</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>  <span class="at">tax_empty=</span><span class="st">&quot;OTU&quot;</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="7_Time_series_files/figure-html/unnamed-chunk-5-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="time-series-with-error-bars" class="section level2">
<h2>Time series with error bars</h2>
<p>Now let’s try reproducing this figure manually to add an error bar
(Standard error). We’ll start again with the code we used for the
stacked bar charts. First, let’s rework the list of predefined species
to match that of the graph obtained with Ampvis2, and replace anything
not in our list with “Other”:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>get_cols <span class="ot">&lt;-</span> <span class="cf">function</span> (n){</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#8dd3c7&quot;</span>, <span class="st">&quot;#ffffb3&quot;</span>, <span class="st">&quot;#bebada&quot;</span>, <span class="st">&quot;#fb8072&quot;</span>, <span class="st">&quot;#80b1d3&quot;</span>,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>           <span class="st">&quot;#fdb462&quot;</span>, <span class="st">&quot;#b3de69&quot;</span>, <span class="st">&quot;#fccde5&quot;</span>, <span class="st">&quot;#d9d9d9&quot;</span>, <span class="st">&quot;#bc80bd&quot;</span>,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>           <span class="st">&quot;#ccebc5&quot;</span>, <span class="st">&quot;#ffed6f&quot;</span>)</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  </span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  col2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#1f78e4&quot;</span>, <span class="st">&quot;#ffff33&quot;</span>, <span class="st">&quot;#c2a5cf&quot;</span>, <span class="st">&quot;#ff7f00&quot;</span>, <span class="st">&quot;#810f7c&quot;</span>,</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>            <span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#006d2c&quot;</span>, <span class="st">&quot;#4d4d4d&quot;</span>, <span class="st">&quot;#8c510a&quot;</span>, <span class="st">&quot;#d73027&quot;</span>,</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>            <span class="st">&quot;#78c679&quot;</span>, <span class="st">&quot;#7f0000&quot;</span>, <span class="st">&quot;#41b6c4&quot;</span>, <span class="st">&quot;#e7298a&quot;</span>, <span class="st">&quot;#54278f&quot;</span>)</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>  </span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>  col3 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#1f78b4&quot;</span>, <span class="st">&quot;#b2df8a&quot;</span>, <span class="st">&quot;#33a02c&quot;</span>, <span class="st">&quot;#fb9a99&quot;</span>,</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>            <span class="st">&quot;#e31a1c&quot;</span>, <span class="st">&quot;#fdbf6f&quot;</span>, <span class="st">&quot;#ff7f00&quot;</span>, <span class="st">&quot;#cab2d6&quot;</span>, <span class="st">&quot;#6a3d9a&quot;</span>,</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>            <span class="st">&quot;#ffff99&quot;</span>, <span class="st">&quot;#b15928&quot;</span>)</span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>  <span class="fu">colorRampPalette</span>(col2)(n)</span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>}</span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>WorkingPS_melt <span class="ot">&lt;-</span> <span class="fu">psmelt</span>(WorkingPS_glom)</span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="fu">head</span>(WorkingPS_melt)</span></code></pre></div>
<pre><code>##                                   OTU          Sample Abundance
## 1632 2949b355cd00039476d9a808f2e76b8e IonT_QC0621WK33    397142
## 1639 2949b355cd00039476d9a808f2e76b8e IonT_QC0421WK33    393894
## 1567 2949b355cd00039476d9a808f2e76b8e IonT_QC0421WK32    342518
## 1649 2949b355cd00039476d9a808f2e76b8e IonT_QC0621WK34    327917
## 1545 2949b355cd00039476d9a808f2e76b8e IonT_QC0521WK33    307748
## 1587 2949b355cd00039476d9a808f2e76b8e IonT_QC0621WK32    270097
##                   InputFileName   Samplename   seq_tech SiteID      Region Year
## 1632 B82_9_L001_R1_001.fastq.gz QC06_21_WK33 IonTorrent   QC06 Sherrington 2021
## 1639 B80_9_L001_R1_001.fastq.gz QC04_21_WK33 IonTorrent   QC04 Napierville 2021
## 1567 B73_9_L001_R1_001.fastq.gz QC04_21_WK32 IonTorrent   QC04 Napierville 2021
## 1649 B89_9_L001_R1_001.fastq.gz QC06_21_WK34 IonTorrent   QC06 Sherrington 2021
## 1545 B81_9_L001_R1_001.fastq.gz QC05_21_WK33 IonTorrent   QC05  St-Edouard 2021
## 1587 B75_9_L001_R1_001.fastq.gz QC06_21_WK32 IonTorrent   QC06 Sherrington 2021
##      Week Week2       Date  X_coord  Y_coord   Kingdom   Phylum         Class
## 1632 WK33    33 2021-08-09 45.13174 -73.4719 Eukaryota Oomycota Stramenopiles
## 1639 WK33    33 2021-08-09 45.18664 -73.3426 Eukaryota Oomycota Stramenopiles
## 1567 WK32    32 2021-08-02 45.18664 -73.3426 Eukaryota Oomycota Stramenopiles
## 1649 WK34    34 2021-08-16 45.13174 -73.4719 Eukaryota Oomycota Stramenopiles
## 1545 WK33    33 2021-08-09 45.19820 -73.5421 Eukaryota Oomycota Stramenopiles
## 1587 WK32    32 2021-08-02 45.13174 -73.4719 Eukaryota Oomycota Stramenopiles
##               Order          Family       Genus                Species
## 1632 Peronosporales Peronosporaceae Peronospora Peronospora manshurica
## 1639 Peronosporales Peronosporaceae Peronospora Peronospora manshurica
## 1567 Peronosporales Peronosporaceae Peronospora Peronospora manshurica
## 1649 Peronosporales Peronosporaceae Peronospora Peronospora manshurica
## 1545 Peronosporales Peronosporaceae Peronospora Peronospora manshurica
## 1587 Peronosporales Peronosporaceae Peronospora Peronospora manshurica</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>predefined_species <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Peronospora manshurica&quot;</span>,</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>                       <span class="st">&quot;Peronospora variabilis&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>                       <span class="st">&quot;Bremia lactucae cluster&quot;</span>,</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora nesliae&quot;</span>,</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>                       <span class="st">&quot;Peronospora destructor&quot;</span>,</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora camelinae&quot;</span>,</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>                       <span class="st">&quot;Albugo candida&quot;</span>,</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>                       <span class="st">&quot;Peronospora aparines&quot;</span>,</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>                       <span class="st">&quot;Peronospora erucastri&quot;</span>,</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>                       <span class="st">&quot;Plasmopara viticola&quot;</span>,</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>                       <span class="st">&quot;Peronospora boni-henrici cluster&quot;</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>)</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>WorkingPS_melt2 <span class="ot">&lt;-</span>  WorkingPS_melt <span class="sc">%&gt;%</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">case_when</span>(</span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>    Species <span class="sc">%in%</span> </span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>      predefined_species <span class="sc">~</span> Species,  <span class="co"># Keep species from the list</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Other&quot;</span>  <span class="co"># Replace other species with &quot;other&quot;</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>  ))</span></code></pre></div>
<p>Now we need to add a column for standard error (SE) in addition to
relative abundance (rel_abund).</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>genus_abundance2 <span class="ot">&lt;-</span> WorkingPS_melt2 <span class="sc">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="fu">group_by</span>(Week2) <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rel_abund =</span> Abundance <span class="sc">/</span> <span class="fu">sum</span>(Abundance)<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SE =</span> <span class="fu">sd</span>(rel_abund)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(rel_abund))) <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Abundance) <span class="sc">%&gt;%</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>),</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">&quot;level&quot;</span>,</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">&quot;taxon&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">taxon =</span> <span class="fu">factor</span>(taxon,</span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>                       <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Peronospora manshurica&quot;</span>,</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>                       <span class="st">&quot;Peronospora variabilis&quot;</span>,</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>                       <span class="st">&quot;Other&quot;</span>,</span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a>                       <span class="st">&quot;Bremia lactucae cluster&quot;</span>,</span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora nesliae&quot;</span>,</span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>                       <span class="st">&quot;Peronospora destructor&quot;</span>,</span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a>                       <span class="st">&quot;Hyaloperonospora camelinae&quot;</span>,</span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a>                       <span class="st">&quot;Albugo candida&quot;</span>,</span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a>                       <span class="st">&quot;Peronospora aparines&quot;</span>,</span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a>                       <span class="st">&quot;Peronospora erucastri&quot;</span>,</span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a>                       <span class="st">&quot;Plasmopara viticola&quot;</span>,</span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a>                       <span class="st">&quot;Peronospora boni-henrici cluster&quot;</span></span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a>                       )))</span></code></pre></div>
<p>Then, summarize the data at the species level and pipe that into
ggplot2 to get our line plot for the predefined species with error
bars.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>genus_abundance2 <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="fu">filter</span>(level<span class="sc">==</span><span class="st">&quot;Species&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  <span class="fu">group_by</span>(Week2, taxon) <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rel_abund =</span> <span class="fu">sum</span>(rel_abund), <span class="at">SE=</span><span class="fu">sum</span>(SE)) <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>  <span class="fu">group_by</span>(Week2, taxon) <span class="sc">%&gt;%</span> </span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_rel_abund=</span><span class="fu">mean</span>(rel_abund), <span class="at">SE =</span> <span class="fu">mean</span>(SE), <span class="at">.groups=</span><span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Week2, <span class="at">y=</span>mean_rel_abund, <span class="at">fill=</span>taxon))<span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>mean_rel_abund<span class="sc">-</span>SE, <span class="at">ymax=</span>mean_rel_abund<span class="sc">+</span>SE), <span class="at">width=</span><span class="fl">0.2</span>)<span class="sc">+</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>  <span class="fu">geom_line</span>()<span class="sc">+</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(taxon), <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">scales=</span><span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Week&quot;</span>,</span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>       <span class="at">y=</span><span class="st">&quot;Relative abundance (%)&quot;</span>)<span class="sc">+</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="7_Time_series_files/figure-html/unnamed-chunk-8-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="alpha-diversity-over-time" class="section level2">
<h2>Alpha diversity over time</h2>
<p>In addition to the distribution of the relative abundance of one or
more species over time, we may also be interested in looking at the
evolution of alpha diversity as a function of time.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;MicrobiotaProcess&quot;</span>)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>alpha_oom <span class="ot">&lt;-</span> <span class="fu">get_alphaindex</span>(WorkingPS_glom)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">as.data.frame</span>(alpha_oom))</span></code></pre></div>
<pre><code>##                 Observe Chao1       ACE   Shannon   Simpson    Pielou
## IonT_QC0221WK20       3  3.00  4.125000 0.7774621 0.4609375 0.7076765
## IonT_QC0221WK22       7  8.00 10.224852 1.7214023 0.7812500 0.8846258
## IonT_QC0221WK23       3  3.00  3.527111 0.8815323 0.5546875 0.8024052
## IonT_QC0221WK24       7  7.25  8.152381 1.8037194 0.8125000 0.9269284
## IonT_QC0221WK25       4  4.00  4.603259 1.1626998 0.6484375 0.8387106
## IonT_QC0221WK26       3  3.00  3.000000 1.0821955 0.6562500 0.9850568
##                              InputFileName   Samplename   seq_tech SiteID
## IonT_QC0221WK20  B1_9_L001_R1_001.fastq.gz QC02_21_WK20 IonTorrent   QC02
## IonT_QC0221WK22  B4_9_L001_R1_001.fastq.gz QC02_21_WK22 IonTorrent   QC02
## IonT_QC0221WK23 B10_9_L001_R1_001.fastq.gz QC02_21_WK23 IonTorrent   QC02
## IonT_QC0221WK24 B16_9_L001_R1_001.fastq.gz QC02_21_WK24 IonTorrent   QC02
## IonT_QC0221WK25 B22_9_L001_R1_001.fastq.gz QC02_21_WK25 IonTorrent   QC02
## IonT_QC0221WK26 B29_9_L001_R1_001.fastq.gz QC02_21_WK26 IonTorrent   QC02
##                       Region Year Week Week2       Date  X_coord  Y_coord
## IonT_QC0221WK20 Frelighsburg 2021 WK20    20 2021-05-10 45.05353 -72.8625
## IonT_QC0221WK22 Frelighsburg 2021 WK22    22 2021-05-24 45.05353 -72.8625
## IonT_QC0221WK23 Frelighsburg 2021 WK23    23 2021-05-31 45.05353 -72.8625
## IonT_QC0221WK24 Frelighsburg 2021 WK24    24 2021-06-07 45.05353 -72.8625
## IonT_QC0221WK25 Frelighsburg 2021 WK25    25 2021-06-14 45.05353 -72.8625
## IonT_QC0221WK26 Frelighsburg 2021 WK26    26 2021-06-21 45.05353 -72.8625</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>Observe_status <span class="ot">&lt;-</span> <span class="fu">ggbox</span>(alpha_oom, </span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>                        <span class="at">geom=</span><span class="st">&quot;boxplot&quot;</span>, </span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>                        <span class="at">factorNames=</span><span class="st">&quot;Week&quot;</span>,   </span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>                        <span class="at">compare =</span> <span class="cn">FALSE</span>,</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>                        <span class="at">testmethod =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>                        <span class="at">signifmap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>                        <span class="at">indexNames=</span><span class="st">&quot;Observe&quot;</span>)<span class="sc">+</span> </span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>  <span class="fu">theme</span>( <span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">colour=</span><span class="cn">NA</span>, <span class="at">fill=</span><span class="st">&quot;grey&quot;</span>))<span class="sc">+</span></span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">get_cols</span>(<span class="dv">16</span>))</span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>Shannon_status<span class="ot">&lt;-</span> <span class="fu">ggbox</span>(alpha_oom, </span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a>                       <span class="at">geom=</span><span class="st">&quot;boxplot&quot;</span>, </span>
<span id="cb13-15"><a href="#cb13-15" tabindex="-1"></a>                       <span class="at">factorNames=</span><span class="st">&quot;Week&quot;</span>,   </span>
<span id="cb13-16"><a href="#cb13-16" tabindex="-1"></a>                       <span class="at">compare =</span> <span class="cn">FALSE</span>, </span>
<span id="cb13-17"><a href="#cb13-17" tabindex="-1"></a>                       <span class="at">testmethod =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb13-18"><a href="#cb13-18" tabindex="-1"></a>                       <span class="at">signifmap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb13-19"><a href="#cb13-19" tabindex="-1"></a>                       <span class="at">indexNames=</span><span class="st">&quot;Shannon&quot;</span>)<span class="sc">+</span> </span>
<span id="cb13-20"><a href="#cb13-20" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;Bottom&quot;</span>)<span class="sc">+</span></span>
<span id="cb13-21"><a href="#cb13-21" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">colour=</span><span class="cn">NA</span>, <span class="at">fill=</span><span class="st">&quot;grey&quot;</span>))<span class="sc">+</span></span>
<span id="cb13-22"><a href="#cb13-22" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">get_cols</span>(<span class="dv">16</span>))</span>
<span id="cb13-23"><a href="#cb13-23" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" tabindex="-1"></a></span>
<span id="cb13-25"><a href="#cb13-25" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb13-26"><a href="#cb13-26" tabindex="-1"></a><span class="fu">ggarrange</span>(<span class="fu">ggarrange</span>(Observe_status,</span>
<span id="cb13-27"><a href="#cb13-27" tabindex="-1"></a>                    Shannon_status,</span>
<span id="cb13-28"><a href="#cb13-28" tabindex="-1"></a>                    <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>),</span>
<span id="cb13-29"><a href="#cb13-29" tabindex="-1"></a>                    <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">2.5</span>, <span class="dv">1</span>)))</span></code></pre></div>
<p><img src="7_Time_series_files/figure-html/unnamed-chunk-9-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
