<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-07-07" />

<title>Introduction</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>










<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R Biovigilance of Phytopathogens Based on a Metabarcoding Approach</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="setup_requirements.html">Workshop Requirements</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part I: Data cleaning and filtering
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1_import_data.html">Importing data into R</a>
    </li>
    <li>
      <a href="2_preprocessing.html">Data cleaning</a>
    </li>
    <li>
      <a href="3_summaries.html">Data filtering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part II -- Comparing sequencing techniques
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4_stacked_barchart.html">stacked bar charts</a>
    </li>
    <li>
      <a href="5_comparison.html">Comparisons</a>
    </li>
  </ul>
</li>
<li>
  <a href="6_Heat_map.html">Beta diversity and comaprison</a>
</li>
<li>
  <a href="7_Time_series.html">Temporal variation</a>
</li>
<li>
  <a href="8_biomarker-discovery.html">Biomarker-discovery</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hvanderheyden/Plant_Canada_2024_workshop.github.io">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Introduction</h1>
<h4 class="date">2024-07-07</h4>

</div>


<p>R Biovigilance of Phytopathogens Based on a Metabarcoding
Approach</p>
<div id="workshop-organization-committee" class="section level2">
<h2>Workshop organization committee</h2>
<p>Hervé Van der Heyden (AAFC - preparation of workshop materials and
website)</p>
<p>Guillaume J. Bilodeau (CFIA - Workshop Instructor)</p>
<p>Wen Chen (AAFC - Workshop organizer)</p>
<p>Teaching assistants:</p>
<p>Shaheen Bibi (Manitoba University - co-instructor), Minuka
Hewapathirana (CFIA - co-instructor) Carol Bvindi (Manitoba
University).</p>
</div>
<div id="description" class="section level2">
<h2>Description</h2>
<p>Metabarcoding combines DNA barcoding with high-throughput sequencing
(HTS) technologies for rapid and high-throughput identification of
multiple species from environmental samples, offering a powerful tool
for biodiversity studies and ecosystem monitoring. It has transformed
our ability to profile complex microbial communities and track plant
pathogens in various environments.</p>
<p>This workshop will provide hands-on experience in metabarcoding-based
community analysis using R, a versatile programming language and
environment for statistical computing and graphics. You will learn about
various R packages and tools that are essential for community data
analysis, enabling you to effectively analyze and interpret
metabarcoding data.</p>
<p>This workshop is designed to demonstrate how you may use
metabarcoding for plant pathogen monitoring and tracking. This is
crucial for early detection and management of plant diseases and for
agriculture and biodiversity conservation. We will explore case studies
and practical applications, highlighting how metabarcoding, combined
with R analysis, becomes a potential diagnostic tool for Biovigilance of
phytopathogens.</p>
</div>
<div id="what-the-workshop-is-not-about" class="section level2">
<h2>What the workshop is not about</h2>
<p>This workshop is not dedicated to learning the basics of R and
R-studio. We believe there are plenty of tutorials and workshops
available online or through research groups. Here are a few examples of
resources for learning the basics:</p>
<p>Sydney E. Everhart, Nikita Gambhir, Kaitlin Gold, Lucky Mehra and
Zhian N. Kamvar offered APS attendees an introduction to R for plant
pathologists in 2020, available at:</p>
<p><a href="https://github.com/everhartlab/APS_IntroR_2020"
class="uri">https://github.com/everhartlab/APS_IntroR_2020</a></p>
<p>Several tutorials and primers are also available from the Grunwald
group. See, for example, this primer on reproducible science using R,
prepared by Zachary S.L. Foster, Zhian N. Kamvar, and Niklaus J.
Grünwald:</p>
<p><a href="https://grunwaldlab.github.io/Reproducible-science-in-R/"
class="uri">https://grunwaldlab.github.io/Reproducible-science-in-R/</a></p>
<p>The Centre québécois des sciences de la biodiversité (CQBS) also
offers a wide range of training courses at various levels, from
introduction classes to advanced multivariate analyses:</p>
<p><a href="https://r.qcbs.ca/workshops/"
class="uri">https://r.qcbs.ca/workshops/</a></p>
<p>Another great source of R tutorials and codes is the Riffomonas code
club page:</p>
<p><a href="https://riffomonas.org/code_club/"
class="uri">https://riffomonas.org/code_club/</a></p>
</div>
<div id="data-used-for-the-workshop" class="section level2">
<h2>Data used for the workshop</h2>
<div class="figure" style="text-align: center">
<img src="Figures/presentation1.jpg" alt="Figure 1. Location of the sites sampled used for this workshop (credit to Philippe Vigneault (AAFC))" width="100%" />
<p class="caption">
Figure 1. Location of the sites sampled used for this workshop (credit
to Philippe Vigneault (AAFC))
</p>
</div>
<p>Data were collected once a week in 2021 in the southwest Montreal
Muck soil region using a Cyclone sampler. DNA extractions were performed
in Dr. Chen’s laboratory and sequenced on Illumina (Chen), IonTorrent
(Bilodeau) and Nanopore (Van der Heyden).</p>
<p>Data were collected by Dr. Carrisse’s team once a week in 2021 in the
Muck’s region southwest of Montreal using a Cyclone sampler (Figure 1).
DNA extractions were performed in Dr. Chen’s laboratory and sequenced on
Illumina (Chen–oomycetes only), IonTorrent (Bilodeau–oomycetes only) and
Nanopore (Van der Heyden–oomycetes and fungi). Illumina and IonTorrent
data were analysed using Dr Marc-Olivier Duceppe’s pipeline (avilable
here: <a href="https://github.com/duceppemo/QIIME2_ITS"
class="uri">https://github.com/duceppemo/QIIME2_ITS</a>) with a database
modified from UNITE v9.0 (2023-07-18), while Nanopore data were analysed
using a custom script based on Minimap2.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
