<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Workshop Requirements</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R Biovigilance of Phytopathogens Based on a Metabarcoding Approach</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="setup_requirements.html">Workshop Requirements</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part I: Data cleaning and filtering
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1_import_data.html">Importing data into R</a>
    </li>
    <li>
      <a href="2_preprocessing.html">Data cleaning</a>
    </li>
    <li>
      <a href="3_summaries.html">Data filtering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Part II -- Comparing sequencing techniques
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4_stacked_barchart.html">stacked bar charts</a>
    </li>
    <li>
      <a href="5_comparison.html">Comparisons</a>
    </li>
  </ul>
</li>
<li>
  <a href="6_Heat_map.html">Beta diversity and comaprison</a>
</li>
<li>
  <a href="7_Time_series.html">Temporal variation</a>
</li>
<li>
  <a href="8_biomarker-discovery.html">Biomarker-discovery</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hvanderheyden/Plant_Canada_2024_workshop.github.io">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Workshop Requirements</h1>

</div>


<div id="installing-r-and-rstudio" class="section level2">
<h2>Installing R and RStudio</h2>
<p>Install R and RStudio:</p>
<ol style="list-style-type: decimal">
<li><p>Download and install the <a href="http://www.r-project.org">R</a>
statistical computing and graphing environment. This works
cross-platform on Windows, OS X and Linux operating systems.</p></li>
<li><p>Download and install the free, open source edition of the <a
href="https://www.rstudio.com/products/rstudio/download/">RStudio
Desktop</a> integrated development environment (IDE), which we
recommend. This is basically a point-and-click interface for R that
includes a text editor, file browser, and some other
conveniences.</p></li>
</ol>
</div>
<div id="required-r-packages" class="section level2">
<h2>Required R packages</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># install.packages(c(&quot;tidyverse&quot;, &quot;ape&quot;, &quot;cowplot&quot;, &quot;ggpubr&quot;, &quot;remotes&quot;, &quot;BiocManager&quot;, &quot;UpSetR&quot;))</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;tidyverse&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ape&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;cowplot&quot;</span>)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;remotes&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;UpSetR&quot;</span>)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co"># BiocManager::install(c(&quot;phyloseq&quot;, &quot;microbiome&quot;,&quot;MicrobiotaProcess&quot;))</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;microbiome&quot;</span>)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;MicrobiotaProcess&quot;</span>)</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="co"># remotes::install_github(&quot;microsud/microbiomeutilities&quot;)</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;microbiomeutilities&quot;</span>) </span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="co"># remotes::install_github(&quot;kasperskytte/ampvis2&quot;)</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ampvis2&quot;</span>) </span></code></pre></div>
</div>
<div id="session-info" class="section level2">
<h2>Session info</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;knitr&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.4.1 (2024-06-14 ucrt)
##  os       Windows 10 x64 (build 19045)
##  system   x86_64, mingw32
##  ui       RTerm
##  language (EN)
##  collate  French_Canada.utf8
##  ctype    French_Canada.utf8
##  tz       America/Toronto
##  date     2024-10-24
##  pandoc   3.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ────────────────────────────────────────────────────────────────────────────
##  ! package              * version   date (UTC) lib source
##    abind                  1.4-8     2024-09-12 [1] CRAN (R 4.4.1)
##    ade4                   1.7-22    2023-02-06 [1] CRAN (R 4.4.0)
##    ampvis2              * 2.8.9     2024-05-31 [1] Github (kasperskytte/ampvis2@930dc58)
##    ape                  * 5.8       2024-04-11 [1] CRAN (R 4.4.0)
##    aplot                  0.2.3     2024-06-17 [1] CRAN (R 4.4.1)
##    backports              1.5.0     2024-05-23 [1] CRAN (R 4.4.0)
##    bibtex                 0.5.1     2023-01-26 [1] CRAN (R 4.4.0)
##    Biobase                2.64.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    BiocGenerics           0.50.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    BiocManager          * 1.30.25   2024-08-28 [1] CRAN (R 4.4.1)
##    biomformat             1.32.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    Biostrings             2.72.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    broom                  1.0.7     2024-09-26 [1] CRAN (R 4.4.1)
##    bslib                  0.8.0     2024-07-29 [1] CRAN (R 4.4.1)
##    cachem                 1.1.0     2024-05-16 [1] CRAN (R 4.4.0)
##    car                    3.1-3     2024-09-27 [1] CRAN (R 4.4.1)
##    carData                3.0-5     2022-01-06 [1] CRAN (R 4.4.0)
##    cli                    3.6.2     2023-12-11 [1] CRAN (R 4.4.0)
##    cluster                2.1.6     2023-12-01 [2] CRAN (R 4.4.1)
##    codetools              0.2-20    2024-03-31 [2] CRAN (R 4.4.1)
##    coin                 * 1.4-3     2023-09-27 [1] CRAN (R 4.4.0)
##    colorspace             2.1-0     2023-01-23 [1] CRAN (R 4.4.0)
##    cowplot              * 1.1.3     2024-01-22 [1] CRAN (R 4.4.0)
##    crayon                 1.5.3     2024-06-20 [1] CRAN (R 4.4.1)
##    data.table             1.15.4    2024-03-30 [1] CRAN (R 4.4.0)
##    DelayedArray           0.30.1    2024-05-08 [1] Bioconductor 3.19 (R 4.4.0)
##    devtools               2.4.5     2022-10-11 [1] CRAN (R 4.4.0)
##    digest                 0.6.35    2024-03-11 [1] CRAN (R 4.4.0)
##    dplyr                * 1.1.4     2023-11-17 [1] CRAN (R 4.4.0)
##    ellipsis               0.3.2     2021-04-29 [1] CRAN (R 4.4.0)
##    evaluate               1.0.0     2024-09-17 [1] CRAN (R 4.4.1)
##    fansi                  1.0.6     2023-12-08 [1] CRAN (R 4.4.0)
##    farver                 2.1.2     2024-05-13 [1] CRAN (R 4.4.0)
##    fastmap                1.2.0     2024-05-15 [1] CRAN (R 4.4.0)
##    fontawesome            0.5.2     2023-08-19 [1] CRAN (R 4.4.0)
##    forcats              * 1.0.0     2023-01-29 [1] CRAN (R 4.4.0)
##    foreach                1.5.2     2022-02-02 [1] CRAN (R 4.4.0)
##    Formula                1.2-5     2023-02-24 [1] CRAN (R 4.4.0)
##    fs                     1.6.4     2024-04-25 [1] CRAN (R 4.4.0)
##    generics               0.1.3     2022-07-05 [1] CRAN (R 4.4.0)
##    GenomeInfoDb           1.40.1    2024-05-24 [1] Bioconductor 3.19 (R 4.4.0)
##    GenomeInfoDbData       1.2.12    2024-05-28 [1] Bioconductor
##    GenomicRanges          1.56.1    2024-06-12 [1] Bioconductor 3.19 (R 4.4.0)
##    ggfun                  0.1.6     2024-08-28 [1] CRAN (R 4.4.1)
##    gghalves               0.1.4     2022-11-20 [1] CRAN (R 4.4.0)
##    ggnewscale             0.5.0     2024-07-19 [1] CRAN (R 4.4.1)
##    ggplot2              * 3.5.1     2024-04-23 [1] CRAN (R 4.4.0)
##    ggplotify              0.1.2     2023-08-09 [1] CRAN (R 4.4.0)
##    ggpubr               * 0.6.0     2023-02-10 [1] CRAN (R 4.4.0)
##    ggrepel                0.9.6     2024-09-07 [1] CRAN (R 4.4.1)
##    ggsignif               0.6.4     2022-10-13 [1] CRAN (R 4.4.0)
##    ggstar                 1.0.4     2022-11-08 [1] CRAN (R 4.4.0)
##    ggtree               * 3.12.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    ggtreeExtra          * 1.14.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    glue                   1.7.0     2024-01-09 [1] CRAN (R 4.4.0)
##    gridExtra              2.3       2017-09-09 [1] CRAN (R 4.4.0)
##    gridGraphics           0.5-1     2020-12-13 [1] CRAN (R 4.4.0)
##    gtable                 0.3.5     2024-04-22 [1] CRAN (R 4.4.0)
##    highr                  0.11      2024-05-26 [1] CRAN (R 4.4.0)
##    hms                    1.1.3     2023-03-21 [1] CRAN (R 4.4.0)
##    htmltools              0.5.8.1   2024-04-04 [1] CRAN (R 4.4.0)
##    htmlwidgets            1.6.4     2023-12-06 [1] CRAN (R 4.4.0)
##    httpuv                 1.6.15    2024-03-26 [1] CRAN (R 4.4.0)
##    httr                   1.4.7     2023-08-15 [1] CRAN (R 4.4.0)
##    igraph                 2.0.3     2024-03-13 [1] CRAN (R 4.4.0)
##    IRanges                2.38.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    iterators              1.0.14    2022-02-05 [1] CRAN (R 4.4.0)
##    jquerylib              0.1.4     2021-04-26 [1] CRAN (R 4.4.0)
##    jsonlite               1.8.8     2023-12-04 [1] CRAN (R 4.4.0)
##    knitcitations        * 1.0.12    2021-01-10 [1] CRAN (R 4.4.0)
##    knitr                * 1.48      2024-07-07 [1] CRAN (R 4.4.1)
##    labeling               0.4.3     2023-08-29 [1] CRAN (R 4.4.0)
##    later                  1.3.2     2023-12-06 [1] CRAN (R 4.4.0)
##    lattice              * 0.22-6    2024-03-20 [2] CRAN (R 4.4.1)
##    lazyeval               0.2.2     2019-03-15 [1] CRAN (R 4.4.0)
##    libcoin                1.0-10    2023-09-27 [1] CRAN (R 4.4.0)
##    lifecycle              1.0.4     2023-11-07 [1] CRAN (R 4.4.0)
##    lubridate            * 1.9.3     2023-09-27 [1] CRAN (R 4.4.0)
##    magrittr               2.0.3     2022-03-30 [1] CRAN (R 4.4.0)
##    MASS                   7.3-60.2  2024-04-26 [2] CRAN (R 4.4.1)
##    Matrix                 1.7-0     2024-04-26 [2] CRAN (R 4.4.1)
##    MatrixGenerics         1.16.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    matrixStats            1.4.1     2024-09-08 [1] CRAN (R 4.4.1)
##    memoise                2.0.1     2021-11-26 [1] CRAN (R 4.4.0)
##    mgcv                   1.9-1     2023-12-21 [2] CRAN (R 4.4.1)
##    microbiome           * 1.26.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    microbiomeutilities  * 1.00.17   2024-05-28 [1] Github (microsud/microbiomeutilities@046a9f9)
##    MicrobiotaProcess    * 1.16.1    2024-07-28 [1] Bioconductor 3.19 (R 4.4.1)
##    mime                   0.12      2021-09-28 [1] CRAN (R 4.4.0)
##    miniUI                 0.1.1.1   2018-05-18 [1] CRAN (R 4.4.0)
##    modeltools             0.2-23    2020-03-05 [1] CRAN (R 4.4.0)
##    multcomp               1.4-26    2024-07-18 [1] CRAN (R 4.4.1)
##    multtest               2.60.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    munsell                0.5.1     2024-04-01 [1] CRAN (R 4.4.0)
##    mvtnorm                1.3-1     2024-09-03 [1] CRAN (R 4.4.1)
##    nlme                   3.1-164   2023-11-27 [2] CRAN (R 4.4.1)
##    patchwork              1.2.0     2024-01-08 [1] CRAN (R 4.4.0)
##    permute              * 0.9-7     2022-01-27 [1] CRAN (R 4.4.0)
##    pheatmap               1.0.12    2019-01-04 [1] CRAN (R 4.4.0)
##    phyloseq             * 1.48.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    pillar                 1.9.0     2023-03-22 [1] CRAN (R 4.4.0)
##    pkgbuild               1.4.4     2024-03-17 [1] CRAN (R 4.4.0)
##    pkgconfig              2.0.3     2019-09-22 [1] CRAN (R 4.4.0)
##    pkgload                1.4.0     2024-06-28 [1] CRAN (R 4.4.1)
##    plotly                 4.10.4    2024-01-13 [1] CRAN (R 4.4.0)
##    plyr                   1.8.9     2023-10-02 [1] CRAN (R 4.4.0)
##    profvis                0.3.8     2023-05-02 [1] CRAN (R 4.4.0)
##    promises               1.3.0     2024-04-05 [1] CRAN (R 4.4.0)
##    purrr                * 1.0.2     2023-08-10 [1] CRAN (R 4.4.0)
##    R6                     2.5.1     2021-08-19 [1] CRAN (R 4.4.0)
##    RColorBrewer           1.1-3     2022-04-03 [1] CRAN (R 4.4.0)
##    Rcpp                   1.0.12    2024-01-09 [1] CRAN (R 4.4.0)
##    readr                * 2.1.5     2024-01-10 [1] CRAN (R 4.4.0)
##    RefManageR             1.4.0     2022-09-30 [1] CRAN (R 4.4.0)
##    remotes              * 2.5.0     2024-03-17 [1] CRAN (R 4.4.0)
##    reshape2               1.4.4     2020-04-09 [1] CRAN (R 4.4.0)
##    rhdf5                  2.48.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##  D rhdf5filters           1.16.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    Rhdf5lib               1.26.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    rlang                  1.1.3     2024-01-10 [1] CRAN (R 4.4.0)
##    rmarkdown              2.28      2024-08-17 [1] CRAN (R 4.4.1)
##    rstatix                0.7.2.999 2024-10-09 [1] Github (kassambara/rstatix@360cda4)
##    rstudioapi             0.16.0    2024-03-24 [1] CRAN (R 4.4.0)
##    Rtsne                  0.17      2023-12-07 [1] CRAN (R 4.4.0)
##    S4Arrays               1.4.1     2024-05-20 [1] Bioconductor 3.19 (R 4.4.0)
##    S4Vectors              0.42.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    sandwich               3.1-0     2023-12-11 [1] CRAN (R 4.4.0)
##    sass                   0.4.9     2024-03-15 [1] CRAN (R 4.4.0)
##    scales                 1.3.0     2023-11-28 [1] CRAN (R 4.4.0)
##    sessioninfo            1.2.2     2021-12-06 [1] CRAN (R 4.4.0)
##    shadowtext             0.1.4     2024-07-18 [1] CRAN (R 4.4.1)
##    shiny                  1.9.1     2024-08-01 [1] CRAN (R 4.4.1)
##    SparseArray            1.4.8     2024-06-02 [1] Bioconductor 3.19 (R 4.4.0)
##    stringi                1.8.4     2024-05-06 [1] CRAN (R 4.4.0)
##    stringr              * 1.5.1     2023-11-14 [1] CRAN (R 4.4.0)
##    SummarizedExperiment   1.34.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    survival             * 3.6-4     2024-04-24 [2] CRAN (R 4.4.1)
##    TH.data                1.1-2     2023-04-17 [1] CRAN (R 4.4.0)
##    tibble               * 3.2.1     2023-03-20 [1] CRAN (R 4.4.0)
##    tidyr                * 1.3.1     2024-01-24 [1] CRAN (R 4.4.0)
##    tidyselect             1.2.1     2024-03-11 [1] CRAN (R 4.4.0)
##    tidytree               0.4.6     2023-12-12 [1] CRAN (R 4.4.0)
##    tidyverse            * 2.0.0     2023-02-22 [1] CRAN (R 4.4.0)
##    timechange             0.3.0     2024-01-18 [1] CRAN (R 4.4.0)
##    treeio                 1.28.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    tzdb                   0.4.0     2023-05-12 [1] CRAN (R 4.4.0)
##    UCSC.utils             1.0.0     2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    UpSetR               * 1.4.0     2019-05-22 [1] CRAN (R 4.4.1)
##    urlchecker             1.0.1     2021-11-30 [1] CRAN (R 4.4.0)
##    usethis                3.0.0     2024-07-29 [1] CRAN (R 4.4.1)
##    utf8                   1.2.4     2023-10-22 [1] CRAN (R 4.4.0)
##    vctrs                  0.6.5     2023-12-01 [1] CRAN (R 4.4.0)
##    vegan                * 2.6-6.1   2024-05-21 [1] CRAN (R 4.4.0)
##    viridisLite            0.4.2     2023-05-02 [1] CRAN (R 4.4.0)
##    withr                  3.0.1     2024-07-31 [1] CRAN (R 4.4.1)
##    xfun                   0.48      2024-10-03 [1] CRAN (R 4.4.1)
##    xml2                   1.3.6     2023-12-04 [1] CRAN (R 4.4.0)
##    xtable                 1.8-4     2019-04-21 [1] CRAN (R 4.4.0)
##    XVector                0.44.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    yaml                   2.3.10    2024-07-26 [1] CRAN (R 4.4.1)
##    yulab.utils            0.1.7     2024-08-26 [1] CRAN (R 4.4.1)
##    zlibbioc               1.50.0    2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
##    zoo                    1.8-12    2023-04-13 [1] CRAN (R 4.4.0)
## 
##  [1] C:/Users/vanderheydenh/AppData/Local/R/win-library/4.4
##  [2] C:/Program Files/R/R-4.4.1/library
## 
##  D ── DLL MD5 mismatch, broken installation.
## 
## ───────────────────────────────────────────────────────────────────────────────────────</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
